---
title: "SGE Analysis Pipeline to run for all regions before merging dfs"
author: "Phoebe Dace, The Francis Crick Institute"
output:
  html_document:
    toc: yes
    df_print: paged
  html_notebook:
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

Initial Setup
==============
#rerun samples in new out_dir
```{r}
#Load packages needed
require(gridExtra)
require(mclust)
library(mixtools)
require(ggplot2)
library(plyr)
require(VennDiagram)
library(gam)
library(plotROC)
require(tidyverse)

cut_pos <- list(
  "x6a" = 41256199,
  "x11" = 41243036,
  "u1" = 41277451,
  "x10a" = 41246799,
  "x10h" = 41244944,
  "x10h2" = 41246349,
  "x12a" = 41234517,
  "x12b" = 41234460,
  "x10d" = 41243499,
  "i11" = 41238873,
  "x5" = 41256965,
  "u2" = 41277462,
  "x1" = 41277356
)

#Need to update all this 
#Do not change replicates throughout script, only here
date <- "220421"
SGE_region <- "x10h"
rep1 <- "rL41"
rep2 <- "rL42"
BRCA1region_cut_pos <- cut_pos[[SGE_region]]

#Load dataset
#load variables
BRCA1_editing_info <- read.table("/Users/dacep/Downloads/Run_SGE_scripts/BRCA1_editing_data.txt",header=TRUE, sep="\t")
#must rename this for each experiment.
experiment_dir <- "/Users/dacep/Downloads/Run_SGE_scripts/pipeline_output/"


#Defining colours. Different colours for different sets being plotted.
named_conseq_colors = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#646CFF",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#66B1A3",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000")

named_conseq_colors_full_simple = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#929000",
  "Likely benign" = "#88CCEE",
  "Likely pathogenic" = "#AA4499",
  "Pathogenic" = "#CC6677",
  "Uncertain significance" = "#326C37",
  "Gray" = "#000000")


```
*Subset Data*
============

```{r}
#switching rep. names to be compatible. Defining the full path then reading it into the table. Use "find and replace" feature to change the region when doing a different analysis. Choose 'all' and tick 'wrap'.
BRCA1regionrL41_file <- paste(experiment_dir,'BRCA1', SGE_region, rep1 ,'_annotated.txt',sep='')
BRCA1regionrL42_file <- paste(experiment_dir,'BRCA1', SGE_region, rep2 ,'_annotated.txt',sep='')
BRCA1regionrL41 <- read.table(BRCA1regionrL41_file, header=TRUE, sep="\t")
BRCA1regionrL42 <- read.table(BRCA1regionrL42_file, header=TRUE, sep="\t")
#Table above shows editing strings, should have one that matches the editing info file near to the top as it's sorted by earliest timepoint reads.
#rename the columns in above table to match code below in place:
BRCA1regionrL41 <- BRCA1regionrL41 %>%
  rename(CADD.phred = PHRED,
    CADD.raw = RawScore,
    pos = Pos,
    conseq = Consequence,
    alt = Alt,
    polyphen.val = PolyPhenVal)

BRCA1regionrL42 <- BRCA1regionrL42 %>%
  rename(CADD.phred = PHRED,
    CADD.raw = RawScore,
    pos = Pos,
    conseq = Consequence,
    alt = Alt,
    polyphen.val = PolyPhenVal)

#SUBSETTING ALL OF THE DATA:
#which p_hdr_read to use? get from editing info file. 'which' function filters out any amplicons that aren't BRCA1region (in this case)
BRCA1_editing_info_BRCA1region <-
  BRCA1_editing_info[which(BRCA1_editing_info$amplicon == paste0('BRCA1', SGE_region)), ]
#HDR_5_SNV  HDR_3_SNV  cut_pos  cut_end  mut_start  mut_end
BRCA1region_HDR_5_SNV = as.character(BRCA1_editing_info_BRCA1region$HDR_5_SNV)
BRCA1region_HDR_3_SNV = as.character(BRCA1_editing_info_BRCA1region$HDR_3_SNV)
#BRCA1region_cut_pos = as.character(BRCA1_editing_info_BRCA1region$cut_pos)
BRCA1region_cut_end = as.character(BRCA1_editing_info_BRCA1region$cut_end)
BRCA1region_mut_start = as.character(BRCA1_editing_info_BRCA1region$mut_start)
BRCA1region_mut_end = as.character(BRCA1_editing_info_BRCA1region$mut_end)
```
*Replicate 1 Analysis*
====================
```{r}
### Replicate 1
BRCA1regionrL41_cSNV_all <- which(BRCA1regionrL41[,'c_hdr_snv']=='True')
BRCA1regionrL41_cSNV_all_df = BRCA1regionrL41[BRCA1regionrL41_cSNV_all,]
BRCA1regionrL41_prog_var_all <- which(BRCA1regionrL41[,'all_hdr_variant']=='True' & BRCA1regionrL41[,'edit_string']!='-WT')
BRCA1regionrL41_prog_var_all_df = BRCA1regionrL41[BRCA1regionrL41_prog_var_all,]
BRCA1regionrL41_prog_map_all = which(BRCA1regionrL41_prog_var_all_df[,'pos']!='REF')
BRCA1regionrL41_prog_map_all_df = BRCA1regionrL41_prog_var_all_df[BRCA1regionrL41_prog_map_all,]
BRCA1regionrL41_prog_map_exon_all = which(BRCA1regionrL41_prog_map_all_df[,'cDNApos']!='REF' & BRCA1regionrL41_prog_map_all_df[,'cDNApos']!='NA')
BRCA1regionrL41_prog_map_exon_all_df = BRCA1regionrL41_prog_map_all_df[BRCA1regionrL41_prog_map_exon_all,]
BRCA1regionrL41_prog_map_CADD = which(BRCA1regionrL41_prog_map_all_df[,'CADD.phred']!='NA')
BRCA1regionrL41_prog_map_CADD_df = BRCA1regionrL41_prog_map_all_df[BRCA1regionrL41_prog_map_CADD,]
#threshold at 1 in 100,000 pre reads called 'thresh5'
BRCA1regionrL41_prog_var_thresh5 <- which(as.numeric(as.character(BRCA1regionrL41_prog_var_all_df[,'pre_freq']))>=0.00001)
BRCA1regionrL41_prog_var_thresh5_df = BRCA1regionrL41_prog_var_all_df[BRCA1regionrL41_prog_var_thresh5,]
BRCA1regionrL41_prog_map_thresh5 <- which(as.numeric(as.character(BRCA1regionrL41_prog_map_all_df[,'pre_freq']))>=0.00001)
BRCA1regionrL41_prog_map_thresh5_df = BRCA1regionrL41_prog_map_all_df[BRCA1regionrL41_prog_map_thresh5,]
BRCA1regionrL41_prog_map_exon_thresh5 <- which(as.numeric(as.character(BRCA1regionrL41_prog_map_exon_all_df[,'pre_freq']))>=0.00001)
BRCA1regionrL41_prog_map_exon_thresh5_df = BRCA1regionrL41_prog_map_exon_all_df[BRCA1regionrL41_prog_map_exon_thresh5,]
BRCA1regionrL41_prog_map_CADD_thresh5 <- which(as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df[,'pre_freq']))>=0.00001)
BRCA1regionrL41_prog_map_CADD_thresh5_df = BRCA1regionrL41_prog_map_CADD_df[BRCA1regionrL41_prog_map_CADD_thresh5,]

# r1 Number of 'prog_var variants' --> all variants in library (i.e. length * 3 + 1)
BRCA1regionrL41_prog_var_all_lib_count = length(BRCA1regionrL41_prog_var_all_df$lib)
BRCA1regionrL41_prog_var_all_lib_count
# r1 Max and min reads for each variant
BRCA1regionrL41_prog_var_all_min_lib_counts = min(BRCA1regionrL41_prog_var_all_df$lib)
BRCA1regionrL41_prog_var_all_min_lib_counts
BRCA1regionrL41_prog_var_all_max_lib_counts = max(BRCA1regionrL41_prog_var_all_df$lib)
BRCA1regionrL41_prog_var_all_max_lib_counts

# r1 Number of 'prog_map variants' --> will exclude the hdr_wt and any PAM_wt variants in sge region (1 or 2)
BRCA1regionrL41_prog_map_all_lib_count = length(BRCA1regionrL41_prog_map_all_df$lib)
BRCA1regionrL41_prog_map_all_lib_count
# r1 Max and min reads for each variant. Shows the number of reads that the variant with the min number of reads has
BRCA1regionrL41_prog_map_all_min_lib_counts = min(BRCA1regionrL41_prog_map_all_df$lib)
BRCA1regionrL41_prog_map_all_min_lib_counts
BRCA1regionrL41_prog_map_all_max_lib_counts = max(BRCA1regionrL41_prog_map_all_df$lib)
BRCA1regionrL41_prog_map_all_max_lib_counts
```
Histogram of Library Counts
---------------------------
```{r}
# r1 histograms of lib counts. Probably filter our the SNVs with read counts near to 0, lowest small peak
BRCA1regionrL41_prog_map_all_hist_lib = ggplot(BRCA1regionrL41_prog_map_all_df,
                                          aes(x = lib,
                                              fill = conseq,
                                              col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.text.x = element_text(size = 20),
    strip.background = element_blank()) +
  theme(legend.position = "bottom") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_prog_map_all_hist_lib
```
Log Coverage of Library Reads Across SGE Region
-------------------------------------
```{r}
# r1 Position vs. lib SNV coverage
BRCA1regionrL41_prog_map_all_lib_pos = ggplot(BRCA1regionrL41_prog_map_all_df,
                                         aes(x = as.numeric(as.character(pos)),
                                             y = log2(lib))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 library reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_prog_map_all_lib_pos
```
Min and Max Read Counts at Day 5
-----------------------
```{r}
# r1 Number of pre 'SNV variants' --> out of all programmed.
BRCA1regionrL41_prog_var_all_pre_count = length(BRCA1regionrL41_prog_var_all_df$pre)
BRCA1regionrL41_prog_var_all_pre_count
# r1 Max and min pre reads for each prog_var
BRCA1regionrL41_prog_var_all_min_pre_counts = min(BRCA1regionrL41_prog_var_all_df$pre)
BRCA1regionrL41_prog_var_all_min_pre_counts
BRCA1regionrL41_prog_var_all_max_pre_counts = max(BRCA1regionrL41_prog_var_all_df$pre)
BRCA1regionrL41_prog_var_all_max_pre_counts
# r1 Number of pre 'SNV variants' --> out of all non-wt variants (2-3 less than all programmed var.)
BRCA1regionrL41_prog_map_all_pre_count = length(BRCA1regionrL41_prog_map_all_df$pre)
BRCA1regionrL41_prog_map_all_pre_count
# r1 Max and min pre reads for each prog_map
BRCA1regionrL41_prog_map_all_min_pre_counts = min(BRCA1regionrL41_prog_map_all_df$pre)
BRCA1regionrL41_prog_map_all_min_pre_counts
BRCA1regionrL41_prog_map_all_max_pre_counts = max(BRCA1regionrL41_prog_map_all_df$pre)
BRCA1regionrL41_prog_map_all_max_pre_counts
```
Distribution of Number of SNVs by Number of Reads at Day 5
-------------------------------
```{r}
# r1 histograms of pre counts prog_map
BRCA1regionrL41_prog_map_all_hist_pre = ggplot(BRCA1regionrL41_prog_map_all_df,
                                          aes(x = pre,
                                              fill = conseq,
                                              col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "bottom") +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.text.x = element_text(size = 20),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_prog_map_all_hist_pre
```
Coverage of Day 5 Reads Across SGE Region
-----------------------------------------
```{r}
# r1 Position vs. pre SNV coverage. Expect there to be a sharp peak in low hundreds. 
BRCA1regionrL41_prog_map_all_pre_pos = ggplot(BRCA1regionrL41_prog_map_all_df,
                                         aes(x = as.numeric(as.character(pos)),
                                             y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_prog_map_all_pre_pos
```
Min and Max Read Counts at Day 14
----------------------
```{r}
# r1 Number of post 'SNV variants'
BRCA1regionrL41_prog_map_all_post_count = length(BRCA1regionrL41_prog_map_all_df$post)
BRCA1regionrL41_prog_map_all_post_count
# r1 Max and min post reads for each prog_map
BRCA1regionrL41_prog_map_all_min_post_counts = min(BRCA1regionrL41_prog_map_all_df$post)
BRCA1regionrL41_prog_map_all_min_post_counts
BRCA1regionrL41_prog_map_all_max_post_counts = max(BRCA1regionrL41_prog_map_all_df$post)
BRCA1regionrL41_prog_map_all_max_post_counts
```
Distribution of Number of SNVs by Number of Reads at Day 14
----------------------------------------------------------
```{r}
# r1 histograms of post counts. Can increase the bins if it's difficult to interpret.
BRCA1regionrL41_prog_map_all_hist_post = ggplot(BRCA1regionrL41_prog_map_all_df,
                                           aes(x = post,
                                               fill = conseq,
                                               col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values =  named_conseq_colors)

BRCA1regionrL41_prog_map_all_hist_post
```
Coverage of Day 14 Reads Across SGE Region
------------------------------------------
```{r}
# r1 Position vs. post SNV coverage
BRCA1regionrL41_prog_map_all_post_pos = ggplot(BRCA1regionrL41_prog_map_all_df,
                                          aes(x = as.numeric(as.character(pos)),
                                              y = log2(post))) +
  geom_point(aes(colour =  conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_prog_map_all_post_pos
```
prog_map_all_hist_tHDR_pre_pseudo_freq
--------------------------------------
```{r}
# r1 Now using tHDR frequencies instead (for RNA):
BRCA1regionrL41_prog_map_all_syn_df <- BRCA1regionrL41_prog_map_all_df[which(BRCA1regionrL41_prog_map_all_df$conseq == 'SYNONYMOUS'),]
BRCA1regionrL41_prog_map_all_syn_tHDR_rna_pre_quarts <- quantile(log2(BRCA1regionrL41_prog_map_all_syn_df$tHDR_rna_pre_ratio))
BRCA1regionrL41_prog_map_all_syn_tHDR_rna_pre_med <- unname(BRCA1regionrL41_prog_map_all_syn_tHDR_rna_pre_quarts[3])
BRCA1regionrL41_prog_map_all_syn_tHDR_rna_post_quarts <- quantile(log2(BRCA1regionrL41_prog_map_all_syn_df$tHDR_rna_post_ratio))
BRCA1regionrL41_prog_map_all_syn_tHDR_rna_post_med <- unname(BRCA1regionrL41_prog_map_all_syn_tHDR_rna_post_quarts[3])

# r1 histograms of tHDR_pre_pseudo_freq counts prog_map
BRCA1regionrL41_prog_map_all_hist_tHDR_pre_pseudo_freq = ggplot(BRCA1regionrL41_prog_map_all_df,
                                                           aes(x = tHDR_pre_pseudo_freq,
                                                               fill = conseq,
                                                               col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Frequency') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_prog_map_all_hist_tHDR_pre_pseudo_freq
```
Total Pseudo HDR Frequency Coverage at Day 5
-------------------
```{r}
# r1 Position vs. tHDR_pre_pseudo_freq SNV coverage
BRCA1regionrL41_prog_map_all_tHDR_pre_pseudo_freq_pos = ggplot(BRCA1regionrL41_prog_map_all_df,
                                                          aes(x = as.numeric(as.character(pos)),
                                                              y = log10(tHDR_pre_pseudo_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-10 tHDR freq') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_prog_map_all_tHDR_pre_pseudo_freq_pos
```
prog_map_all_hist_tHDR_post_pseudo_freq
-------------------
```{r}
# r1 histograms of tHDR_post_pseudo_freq counts
BRCA1regionrL41_prog_map_all_hist_tHDR_post_pseudo_freq = ggplot(BRCA1regionrL41_prog_map_all_df,
                                                            aes(x = tHDR_post_pseudo_freq,
                                                                fill = conseq,
                                                                col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_prog_map_all_hist_tHDR_post_pseudo_freq
```
Total Pseudo HDR Frequency Coverage at Day 14
----------------------------------------------
```{r}
# r1 Position vs. tHDR_post_pseudo_freq SNV coverage
BRCA1regionrL41_prog_map_all_tHDR_post_pseudo_freq_pos = ggplot(BRCA1regionrL41_prog_map_all_df,
                                                           aes(x = as.numeric(as.character(pos)),
                                                               y = log10(tHDR_post_pseudo_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-10 tHDR_post_pseudo_freq reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_prog_map_all_tHDR_post_pseudo_freq_pos
```
tHDR Read Counts
-----------
```{r}
# r1 Max and min tHDR_rna reads for each prog_map
BRCA1regionrL41_prog_map_all_min_rna_counts = min(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna)
BRCA1regionrL41_prog_map_all_min_rna_counts
BRCA1regionrL41_prog_map_all_max_rna_counts = max(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna)
BRCA1regionrL41_prog_map_all_max_rna_counts
BRCA1regionrL41_prog_map_all_min_rna_pseudo_freq = min(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_pseudo_freq)
BRCA1regionrL41_prog_map_all_min_rna_pseudo_freq
BRCA1regionrL41_prog_map_all_max_rna_pseudo_freq = max(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_pseudo_freq)
BRCA1regionrL41_prog_map_all_max_rna_pseudo_freq
BRCA1regionrL41_prog_map_all_min_rna_pre_ratio = min(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_pre_ratio)
BRCA1regionrL41_prog_map_all_min_rna_pre_ratio
BRCA1regionrL41_prog_map_all_max_rna_pre_ratio = max(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_pre_ratio)
BRCA1regionrL41_prog_map_all_max_rna_pre_ratio
BRCA1regionrL41_prog_map_all_min_rna_post_ratio = min(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_post_ratio)
BRCA1regionrL41_prog_map_all_min_rna_post_ratio
BRCA1regionrL41_prog_map_all_max_rna_post_ratio = max(BRCA1regionrL41_prog_map_exon_all_df$tHDR_rna_post_ratio)
BRCA1regionrL41_prog_map_all_max_rna_post_ratio
```
Correlation Between Samples
---------------------------
```{r}
# r1 check correlation between different experimental samples:
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_rna_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_rna_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_rna_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL41_prog_map_all_df$post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pre_ratio),
  log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_lib_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$post_pre_ratio),
  log2(BRCA1regionrL41_prog_map_all_df$post_lib_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_pre_ratio),
  log2(BRCA1regionrL41_prog_map_all_df$post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_all_df$tHDR_post_lib_ratio),
  log2(BRCA1regionrL41_prog_map_all_df$post_lib_ratio),
  method = 'spearman')

# r1 switching to the df with CADD scores to make correlations with CADD
cor(log2(BRCA1regionrL41_prog_map_CADD_df$post_pre_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_post_pre_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_CADD_df$post_lib_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_post_lib_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_CADD_df$pre_lib_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_pre_lib_ratio),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
```
Sequencing Error Correction: cSNV Variants
----------------------------
```{r}
# r1 # r1 SEQUENCING ERROR CORRECTION ANALYSIS:
# r1 Number of corr_pre 'SNV variants' --> length * 3 snvs minus 3 * PAM sites
BRCA1regionrL41_cSNV_all_corr_pre_count = length(BRCA1regionrL41_cSNV_all_df$corr_pre)
BRCA1regionrL41_cSNV_all_corr_pre_count
# r1 Max and min corr_pre reads for each prog_map
BRCA1regionrL41_cSNV_all_min_corr_pre_counts = min(BRCA1regionrL41_cSNV_all_df$corr_pre)
BRCA1regionrL41_cSNV_all_min_corr_pre_counts
BRCA1regionrL41_cSNV_all_max_corr_pre_counts = max(BRCA1regionrL41_cSNV_all_df$corr_pre)
BRCA1regionrL41_cSNV_all_max_corr_pre_counts
# r1 pre counts vs. corr_pre counts:
BRCA1regionrL41_cSNV_all_pre_corr_pre = ggplot(BRCA1regionrL41_cSNV_all_df,
                                          aes(x = pre,
                                              y = corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. pre reads') +
  xlab('pre reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_cSNV_all_pre_corr_pre
```
cSNV Corrected Day 5 Read Counts r1
------------
```{r}
# r1 histogram of corr_pre counts cSNV
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = corr_pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 position vs. corr_pre SNV coverage
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 pre minus corr_pre by position
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 pre minus corr_pre by position
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq,
                 size = (1 - BRCA1regionrL41_cSNV_all_df$corr_pre / BRCA1regionrL41_cSNV_all_df$pre))) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 Number of corr_post 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_cSNV_all_corr_post_count = length(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_cSNV_all_corr_post_count

# r1 Max and min corr_post reads for each cSNV
BRCA1regionrL41_cSNV_all_min_corr_post_counts = min(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_cSNV_all_min_corr_post_counts
BRCA1regionrL41_cSNV_all_max_corr_post_counts = max(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_cSNV_all_max_corr_post_counts

# r1 post counts vs. corr_post counts:
BRCA1regionrL41_cSNV_all_post_corr_post = ggplot(BRCA1regionrL41_cSNV_all_df,
                                            aes(x = post,
                                                y = corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. post reads') +
  xlab('post reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_cSNV_all_post_corr_post
```

```{r}
# r1 histogram of corr_post counts cSNV
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = corr_post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 Position vs. corr_post SNV coverage
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 post minus corr_post by position
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 post minus corr_post by position
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq,
                 size = (1 - corr_post / post))) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 apply threshold to determined by frequency OR somehow by error correction for low counts...
# r1 looking for a dramatic drop here
ggplot(BRCA1regionrL41_cSNV_all_df,
       aes(x = pre,
           y = corr_pre / pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Fraction true SNVs') +
  xlab('Uncorrected D6 reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_abline(intercept = 1,
              slope = 0,
              lty = 2)
```
Sequencing Error Correction: Partial SNV Variants
-------------------
```{r}
# r1 repeat with pSNV variants (only cut end is edited!)
BRCA1region_p_hdr_snv = paste('p_hdr_',BRCA1region_cut_end,'_snv',sep='')
BRCA1regionrL41_pSNV_all <- which(BRCA1regionrL41[,BRCA1region_p_hdr_snv]=='True')
BRCA1regionrL41_pSNV_all_df = BRCA1regionrL41[BRCA1regionrL41_pSNV_all,]
# r1 Number of 'pSNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_pSNV_all_lib_count = length(BRCA1regionrL41_pSNV_all_df$lib)
BRCA1regionrL41_pSNV_all_lib_count
# r1 Max and min reads for each variant
BRCA1regionrL41_pSNV_all_min_lib_counts = min(BRCA1regionrL41_pSNV_all_df$lib)
BRCA1regionrL41_pSNV_all_min_lib_counts
BRCA1regionrL41_pSNV_all_max_lib_counts = max(BRCA1regionrL41_pSNV_all_df$lib)
BRCA1regionrL41_pSNV_all_max_lib_counts
# r1 Number of pre 'SNV variants'
BRCA1regionrL41_pSNV_all_pre_count = length(BRCA1regionrL41_pSNV_all_df$pre)
BRCA1regionrL41_pSNV_all_pre_count
# r1 Max and min pre reads for each pSNV
BRCA1regionrL41_pSNV_all_min_pre_counts = min(BRCA1regionrL41_pSNV_all_df$pre)
BRCA1regionrL41_pSNV_all_min_pre_counts
BRCA1regionrL41_pSNV_all_max_pre_counts = max(BRCA1regionrL41_pSNV_all_df$pre)
BRCA1regionrL41_pSNV_all_max_pre_counts
# r1 histograms of pre counts pSNV
ggplot(BRCA1regionrL41_pSNV_all_df,
       aes(x = pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 Position vs. pre SNV coverage
ggplot(BRCA1regionrL41_pSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 Number of post 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_pSNV_all_post_count = length(BRCA1regionrL41_pSNV_all_df$post)
BRCA1regionrL41_pSNV_all_post_count
# r1 Max and min post reads for each pSNV
BRCA1regionrL41_pSNV_all_min_post_counts = min(BRCA1regionrL41_pSNV_all_df$post)
BRCA1regionrL41_pSNV_all_min_post_counts
BRCA1regionrL41_pSNV_all_max_post_counts = max(BRCA1regionrL41_pSNV_all_df$post)
BRCA1regionrL41_pSNV_all_max_post_counts
# r1 histograms of post counts
ggplot(BRCA1regionrL41_pSNV_all_df,
       aes(x = post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 Position vs. post SNV coverage
ggplot(BRCA1regionrL41_pSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```
Thresholded Data
------------------
# r1 Plots for thresholded data ('thresh5' indicates 1 in 1e5 or higher freq.):
```{r}
# r1 Number of 'cSNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_prog_map_thresh5_lib_count = length(BRCA1regionrL41_prog_map_thresh5_df$lib)
BRCA1regionrL41_prog_map_thresh5_lib_count
BRCA1regionrL41_prog_map_thresh5_min_lib_counts = min(BRCA1regionrL41_prog_map_thresh5_df$lib)
BRCA1regionrL41_prog_map_thresh5_min_lib_counts
BRCA1regionrL41_prog_map_thresh5_max_lib_counts = max(BRCA1regionrL41_prog_map_thresh5_df$lib)
BRCA1regionrL41_prog_map_thresh5_max_lib_counts
# r1 histograms of lib counts
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = lib,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  theme(legend.position = "none") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 Position vs. lib SNV coverage
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(lib))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 library reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r1 Number of pre 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_prog_map_thresh5_pre_count = length(BRCA1regionrL41_prog_map_thresh5_df$pre)
BRCA1regionrL41_prog_map_thresh5_pre_count
BRCA1regionrL41_prog_map_thresh5_min_pre_counts = min(BRCA1regionrL41_prog_map_thresh5_df$pre)
BRCA1regionrL41_prog_map_thresh5_min_pre_counts
BRCA1regionrL41_prog_map_thresh5_max_pre_counts = max(BRCA1regionrL41_prog_map_thresh5_df$pre)
BRCA1regionrL41_prog_map_thresh5_max_pre_counts

# r1 pre counts vs. corr_pre counts:
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = pre,
           y = corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. pre reads') +
  xlab('pre reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r1 histogram of corr_pre counts prog_map
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = corr_pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r1 Position vs. pre SNV coverage
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 Position vs. corr_pre SNV coverage
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 pre minus corr_pre by position
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 pre minus corr_pre by position with size
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq,
                 size = (1 - BRCA1regionrL41_prog_map_thresh5_df$corr_pre / BRCA1regionrL41_prog_map_thresh5_df$pre))) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 Number of post 'SNV variants' --> will be 2 less than SGE region length x 3 due to PAM edits
BRCA1regionrL41_prog_map_thresh5_post_count = length(BRCA1regionrL41_prog_map_thresh5_df$post)
BRCA1regionrL41_prog_map_thresh5_post_count
BRCA1regionrL41_prog_map_thresh5_min_post_counts = min(BRCA1regionrL41_prog_map_thresh5_df$post)
BRCA1regionrL41_prog_map_thresh5_min_post_counts
BRCA1regionrL41_prog_map_thresh5_max_post_counts = max(BRCA1regionrL41_prog_map_thresh5_df$post)
BRCA1regionrL41_prog_map_thresh5_max_post_counts
BRCA1regionrL41_prog_map_thresh5_corr_post_count = length(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_prog_map_thresh5_corr_post_count
BRCA1regionrL41_prog_map_thresh5_min_corr_post_counts = min(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_prog_map_thresh5_min_corr_post_counts
BRCA1regionrL41_prog_map_thresh5_max_corr_post_counts = max(BRCA1regionrL41_cSNV_all_df$corr_post)
BRCA1regionrL41_prog_map_thresh5_max_corr_post_counts
```

```{r}
# r1 post counts vs. corr_post counts:
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = post,
           y = corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. post reads') +
  xlab('post reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r1 histograms of post counts
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r1 histogram of corr_post counts prog_map
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = corr_post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values =named_conseq_colors)

# r1 Position vs. post SNV coverage
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 Position vs. corr_post SNV coverage
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 post minus corr_post by position
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1, size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
       axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 post minus corr_post by position
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq,
                 size = (1 - BRCA1regionrL41_prog_map_thresh5_df$corr_post / BRCA1regionrL41_prog_map_thresh5_df$post))) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r1 looking for a dramatic drop here
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = pre,
           y = corr_pre / pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr_pre/pre') +
  xlab('pre reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_abline(intercept = 1,
              slope = 0,
              lty = 2)

BRCA1regionrL41_pSNV_thresh5 <-
  which(BRCA1regionrL41[, BRCA1region_p_hdr_snv] == 'True' &
          as.numeric(as.character(BRCA1regionrL41[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL41_pSNV_thresh5_df = BRCA1regionrL41[BRCA1regionrL41_pSNV_thresh5,]
BRCA1regionrL41_pSNV_thresh5_lib_count = length(BRCA1regionrL41_pSNV_thresh5_df$lib)
BRCA1regionrL41_pSNV_thresh5_lib_count
BRCA1regionrL41_pSNV_thresh5_min_lib_counts = min(BRCA1regionrL41_pSNV_thresh5_df$lib)
BRCA1regionrL41_pSNV_thresh5_min_lib_counts
BRCA1regionrL41_pSNV_thresh5_max_lib_counts = max(BRCA1regionrL41_pSNV_thresh5_df$lib)
BRCA1regionrL41_pSNV_thresh5_max_lib_counts
BRCA1regionrL41_pSNV_thresh5_pre_count = length(BRCA1regionrL41_pSNV_thresh5_df$pre)
BRCA1regionrL41_pSNV_thresh5_pre_count
BRCA1regionrL41_pSNV_thresh5_min_pre_counts = min(BRCA1regionrL41_pSNV_thresh5_df$pre)
BRCA1regionrL41_pSNV_thresh5_min_pre_counts
BRCA1regionrL41_pSNV_thresh5_max_pre_counts = max(BRCA1regionrL41_pSNV_thresh5_df$pre)
BRCA1regionrL41_pSNV_thresh5_max_pre_counts
```

```{r}
# r1 histograms of pre counts pSNV
ggplot(BRCA1regionrL41_pSNV_thresh5_df,
       aes(x = pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r1 Position vs. pre pSNV coverage
ggplot(BRCA1regionrL41_pSNV_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1, size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41_pSNV_thresh5_post_count = length(BRCA1regionrL41_pSNV_thresh5_df$post)
BRCA1regionrL41_pSNV_thresh5_post_count
BRCA1regionrL41_pSNV_thresh5_min_post_counts = min(BRCA1regionrL41_pSNV_thresh5_df$post)
BRCA1regionrL41_pSNV_thresh5_min_post_counts
BRCA1regionrL41_pSNV_thresh5_max_post_counts = max(BRCA1regionrL41_pSNV_thresh5_df$post)
BRCA1regionrL41_pSNV_thresh5_max_post_counts
```

```{r}
# r1 histograms of post counts
ggplot(BRCA1regionrL41_pSNV_thresh5_df,
       aes(x = post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r1 Position vs. post SNV coverage
ggplot(BRCA1regionrL41_pSNV_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(post))) +
  geom_point(aes(colour = conseq),
             alpha = 1, size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```
*Replicate 2 Analysis*
====================
```{r}
### Replicate 2
BRCA1regionrL42_cSNV_all <- which(BRCA1regionrL42[, 'c_hdr_snv'] == 'True')
BRCA1regionrL42_cSNV_all_df = BRCA1regionrL42[BRCA1regionrL42_cSNV_all,]
BRCA1regionrL42_prog_var_all <-
  which(BRCA1regionrL42[, 'all_hdr_variant'] == 'True' &
          BRCA1regionrL42[, 'edit_string'] != '-WT')
BRCA1regionrL42_prog_var_all_df = BRCA1regionrL42[BRCA1regionrL42_prog_var_all,]
BRCA1regionrL42_prog_map_all = which(BRCA1regionrL42_prog_var_all_df[, 'pos'] != 'REF')
BRCA1regionrL42_prog_map_all_df = BRCA1regionrL42_prog_var_all_df[BRCA1regionrL42_prog_map_all,]
BRCA1regionrL42_prog_map_exon_all = which(BRCA1regionrL42_prog_map_all_df[, 'cDNApos'] !=
                                       'REF' &
                                       BRCA1regionrL42_prog_map_all_df[, 'cDNApos'] != 'NA')
BRCA1regionrL42_prog_map_exon_all_df = BRCA1regionrL42_prog_map_all_df[BRCA1regionrL42_prog_map_exon_all,]
BRCA1regionrL42_prog_map_CADD = which(BRCA1regionrL42_prog_map_all_df[, 'CADD.phred'] !=
                                   'NA')
BRCA1regionrL42_prog_map_CADD_df = BRCA1regionrL42_prog_map_all_df[BRCA1regionrL42_prog_map_CADD,]

#threshold at 1 in 100,000 pre reads called 'thresh5'
BRCA1regionrL42_prog_var_thresh5 <-
  which(as.numeric(as.character(BRCA1regionrL42_prog_var_all_df[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL42_prog_var_thresh5_df = BRCA1regionrL42_prog_var_all_df[BRCA1regionrL42_prog_var_thresh5,]
BRCA1regionrL42_prog_map_thresh5 <-
  which(as.numeric(as.character(BRCA1regionrL42_prog_map_all_df[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL42_prog_map_thresh5_df = BRCA1regionrL42_prog_map_all_df[BRCA1regionrL42_prog_map_thresh5,]
BRCA1regionrL42_prog_map_exon_thresh5 <-
  which(as.numeric(as.character(BRCA1regionrL42_prog_map_exon_all_df[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL42_prog_map_exon_thresh5_df = BRCA1regionrL42_prog_map_exon_all_df[BRCA1regionrL42_prog_map_exon_thresh5,]
BRCA1regionrL42_prog_map_CADD_thresh5 <-
  which(as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL42_prog_map_CADD_thresh5_df = BRCA1regionrL42_prog_map_CADD_df[BRCA1regionrL42_prog_map_CADD_thresh5,]

# r2 Number of 'prog_var variants' --> all variants in library (i.e. length * 3 + 1)
BRCA1regionrL42_prog_var_all_lib_count = length(BRCA1regionrL42_prog_var_all_df$lib)
BRCA1regionrL42_prog_var_all_lib_count
# r2 Max and min reads for each variant
BRCA1regionrL42_prog_var_all_min_lib_counts = min(BRCA1regionrL42_prog_var_all_df$lib)
BRCA1regionrL42_prog_var_all_min_lib_counts
BRCA1regionrL42_prog_var_all_max_lib_counts = max(BRCA1regionrL42_prog_var_all_df$lib)
BRCA1regionrL42_prog_var_all_max_lib_counts

# r2 Number of 'prog_map variants' --> will exclude the hdr_wt and any PAM_wt variants in sge region (1 or 2)
BRCA1regionrL42_prog_map_all_lib_count = length(BRCA1regionrL42_prog_map_all_df$lib)
BRCA1regionrL42_prog_map_all_lib_count
# r2 Max and min reads for each variant
BRCA1regionrL42_prog_map_all_min_lib_counts = min(BRCA1regionrL42_prog_map_all_df$lib)
BRCA1regionrL42_prog_map_all_min_lib_counts
BRCA1regionrL42_prog_map_all_max_lib_counts = max(BRCA1regionrL42_prog_map_all_df$lib)
BRCA1regionrL42_prog_map_all_max_lib_counts
```
Histogram of Library Counts
--------------
```{r}
# r2 histograms of lib counts
BRCA1regionrL42_prog_map_all_hist_lib = ggplot(BRCA1regionrL42_prog_map_all_df,
                                          aes(x = lib,
                                              fill = conseq,
                                              col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  theme(legend.position = "bottom") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_prog_map_all_hist_lib
```
Log Read Coverage of Library Across SGE Region
-----------------------------------------------
```{r}
# r2 Position vs. lib SNV coverage
BRCA1regionrL42_prog_map_all_lib_pos = ggplot(BRCA1regionrL42_prog_map_all_df,
                                         aes(x = as.numeric(as.character(pos)), 
                                             y = log2(lib))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 library reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL42_prog_map_all_lib_pos
```
Min and Max Reads at Day 5
--------------------------
```{r}
# r2 Number of pre 'SNV variants' --> out of all programmed.
BRCA1regionrL42_prog_var_all_pre_count = length(BRCA1regionrL42_prog_var_all_df$pre)
BRCA1regionrL42_prog_var_all_pre_count
# r2 Max and min pre reads for each prog_var
BRCA1regionrL42_prog_var_all_min_pre_counts = min(BRCA1regionrL42_prog_var_all_df$pre)
BRCA1regionrL42_prog_var_all_min_pre_counts
BRCA1regionrL42_prog_var_all_max_pre_counts = max(BRCA1regionrL42_prog_var_all_df$pre)
BRCA1regionrL42_prog_var_all_max_pre_counts

# r2 Number of pre 'SNV variants' --> out of all non-wt variants (2-3 less than all programmed var.)
BRCA1regionrL42_prog_map_all_pre_count = length(BRCA1regionrL42_prog_map_all_df$pre)
BRCA1regionrL42_prog_map_all_pre_count
# r2 Max and min pre reads for each prog_map
BRCA1regionrL42_prog_map_all_min_pre_counts = min(BRCA1regionrL42_prog_map_all_df$pre)
BRCA1regionrL42_prog_map_all_min_pre_counts
BRCA1regionrL42_prog_map_all_max_pre_counts = max(BRCA1regionrL42_prog_map_all_df$pre)
BRCA1regionrL42_prog_map_all_max_pre_counts
```
Distribution of Number of SNVs by Number of Reads at Day 5
--------------------------------
```{r}
# r2 histograms of pre counts prog_map
BRCA1regionrL42_prog_map_all_hist_pre = ggplot(BRCA1regionrL42_prog_map_all_df,
                                          aes(x = pre,
                                              fill = conseq,
                                              col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_prog_map_all_hist_pre
```
Coverage of Day 5 Reads Across SGE Region
---------------------
```{r}
# r2 Position vs. pre SNV coverage
BRCA1regionrL42_prog_map_all_pre_pos = ggplot(BRCA1regionrL42_prog_map_all_df,
                                         aes(x = as.numeric(as.character(pos)),
                                             y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL42_prog_map_all_pre_pos
```
Min and Max Reads at Day 14
---------------------
```{r}
# r2 Number of post 'SNV variants'
BRCA1regionrL42_prog_map_all_post_count = length(BRCA1regionrL42_prog_map_all_df$post)
BRCA1regionrL42_prog_map_all_post_count

# r2 Max and min post reads for each prog_map
BRCA1regionrL42_prog_map_all_min_post_counts = min(BRCA1regionrL42_prog_map_all_df$post)
BRCA1regionrL42_prog_map_all_min_post_counts
BRCA1regionrL42_prog_map_all_max_post_counts = max(BRCA1regionrL42_prog_map_all_df$post)
BRCA1regionrL42_prog_map_all_max_post_counts
```
Distribution of Number of SNVs by Number of Reads at Day 14
---------------------
```{r}
# r2 histograms of post counts
BRCA1regionrL42_prog_map_all_hist_post = ggplot(BRCA1regionrL42_prog_map_all_df,
                                           aes(x = post,
                                               fill = conseq,
                                               col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_prog_map_all_hist_post
```
Coverage of Day 14 Reads Across SGE Region
---------------------
```{r}
# r2 Position vs. post SNV coverage
BRCA1regionrL42_prog_map_all_post_pos = ggplot(BRCA1regionrL42_prog_map_all_df,
                                          aes(x = as.numeric(as.character(pos)),
                                              y = log2(post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL42_prog_map_all_post_pos
```
Day 5 Total HDR Frequency vs. SNVs
---------------------
```{R}
# r2 Now using tHDR frequencies instead (for RNA):
BRCA1regionrL42_prog_map_all_syn_df <-
  BRCA1regionrL42_prog_map_all_df[which(BRCA1regionrL42_prog_map_all_df$conseq == 'SYNONYMOUS'),]
BRCA1regionrL42_prog_map_all_syn_tHDR_rna_pre_quarts <-
  quantile(log2(BRCA1regionrL42_prog_map_all_syn_df$tHDR_rna_pre_ratio))
BRCA1regionrL42_prog_map_all_syn_tHDR_rna_pre_med <-
  unname(BRCA1regionrL42_prog_map_all_syn_tHDR_rna_pre_quarts[3])
BRCA1regionrL42_prog_map_all_syn_tHDR_rna_post_quarts <-
  quantile(log2(BRCA1regionrL42_prog_map_all_syn_df$tHDR_rna_post_ratio))
BRCA1regionrL42_prog_map_all_syn_tHDR_rna_post_med <-
  unname(BRCA1regionrL42_prog_map_all_syn_tHDR_rna_post_quarts[3])

# r2 histograms of tHDR_pre_pseudo_freq counts prog_map
BRCA1regionrL42_prog_map_all_hist_tHDR_pre_pseudo_freq = ggplot(BRCA1regionrL42_prog_map_all_df,
                                                           aes(x = tHDR_pre_pseudo_freq,
                                                               fill = conseq,
                                                               col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Frequency') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
       axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_prog_map_all_hist_tHDR_pre_pseudo_freq
```
Day 5 Total HDR Frequency vs. Position
---------------------
```{r}
# r2 Position vs. tHDR_pre_pseudo_freq SNV coverage
BRCA1regionrL42_prog_map_all_tHDR_pre_pseudo_freq_pos = ggplot(BRCA1regionrL42_prog_map_all_df,
                                                          aes(x = as.numeric(as.character(pos)),
                                                              y = log10(tHDR_pre_pseudo_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-10 tHDR freq') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL42_prog_map_all_tHDR_pre_pseudo_freq_pos
```
Day 14 Total HDR Frequency vs. SNVs
---------------------
```{r}
# r2 histograms of tHDR_post_pseudo_freq counts
BRCA1regionrL42_prog_map_all_hist_tHDR_post_pseudo_freq = ggplot(BRCA1regionrL42_prog_map_all_df,
                                                            aes(x = tHDR_post_pseudo_freq,
                                                                fill = conseq,
                                                                col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_prog_map_all_hist_tHDR_post_pseudo_freq
```
Day 14 Total HDR Frequency vs. Position
---------------------
```{r}
# r2 Position vs. tHDR_post_pseudo_freq SNV coverage
BRCA1regionrL42_prog_map_all_tHDR_post_pseudo_freq_pos = ggplot(BRCA1regionrL42_prog_map_all_df,
                                                           aes(x = as.numeric(as.character(pos)),
                                                               y = log10(tHDR_post_pseudo_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-10 tHDR_post_pseudo_freq reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
       axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL42_prog_map_all_tHDR_post_pseudo_freq_pos
```
Max and Min tHDR Read Counts Day 13
----------------------------------
```{r}
# r2 Max and min tHDR_rna reads for each prog_map
BRCA1regionrL42_prog_map_all_min_rna_counts = min(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna)
BRCA1regionrL42_prog_map_all_min_rna_counts
BRCA1regionrL42_prog_map_all_max_rna_counts = max(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna)
BRCA1regionrL42_prog_map_all_max_rna_counts
BRCA1regionrL42_prog_map_all_min_rna_pseudo_freq = min(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_pseudo_freq)
BRCA1regionrL42_prog_map_all_min_rna_pseudo_freq
BRCA1regionrL42_prog_map_all_max_rna_pseudo_freq = max(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_pseudo_freq)
BRCA1regionrL42_prog_map_all_max_rna_pseudo_freq
BRCA1regionrL42_prog_map_all_min_rna_pre_ratio = min(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_pre_ratio)
BRCA1regionrL42_prog_map_all_min_rna_pre_ratio
BRCA1regionrL42_prog_map_all_max_rna_pre_ratio = max(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_pre_ratio)
BRCA1regionrL42_prog_map_all_max_rna_pre_ratio
BRCA1regionrL42_prog_map_all_min_rna_post_ratio = min(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_post_ratio)
BRCA1regionrL42_prog_map_all_min_rna_post_ratio
BRCA1regionrL42_prog_map_all_max_rna_post_ratio = max(BRCA1regionrL42_prog_map_exon_all_df$tHDR_rna_post_ratio)
BRCA1regionrL42_prog_map_all_max_rna_post_ratio
```
Correlations Between Samples
----------------------------------
```{r}
# r2 check correlation between different experimental samples:
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_rna_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_rna_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_pre_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_rna_pseudo_freq),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_lib_pseudo_freq),
  log2(BRCA1regionrL42_prog_map_all_df$post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pre_ratio),
  log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_lib_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$post_pre_ratio),
  log2(BRCA1regionrL42_prog_map_all_df$post_lib_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_pre_ratio),
  log2(BRCA1regionrL42_prog_map_all_df$post_pre_ratio),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_all_df$tHDR_post_lib_ratio),
  log2(BRCA1regionrL42_prog_map_all_df$post_lib_ratio),
  method = 'spearman')

# r2 switching to the df with CADD scores to make correlations with CADD
cor(log2(BRCA1regionrL42_prog_map_CADD_df$post_pre_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_post_pre_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_CADD_df$post_lib_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_post_lib_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_CADD_df$pre_lib_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_pre_lib_ratio),
  as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
  method = 'spearman')
```
Sequencing Error Correction Analysis: cSNV Variants
-------------------------------
```{r}
# r2 # r2 SEQUENCING ERROR CORRECTION ANALYSIS:
# r2 Number of corr_pre 'SNV variants' --> length * 3 snvs minus 3 * PAM sites
BRCA1regionrL42_cSNV_all_corr_pre_count = length(BRCA1regionrL42_cSNV_all_df$corr_pre)
BRCA1regionrL42_cSNV_all_corr_pre_count
# r2 Max and min corr_pre reads for each prog_map
BRCA1regionrL42_cSNV_all_min_corr_pre_counts = min(BRCA1regionrL42_cSNV_all_df$corr_pre)
BRCA1regionrL42_cSNV_all_min_corr_pre_counts
BRCA1regionrL42_cSNV_all_max_corr_pre_counts = max(BRCA1regionrL42_cSNV_all_df$corr_pre)
BRCA1regionrL42_cSNV_all_max_corr_pre_counts

# r2 pre counts vs. corr_pre counts:
BRCA1regionrL42_cSNV_all_pre_corr_pre = ggplot(BRCA1regionrL42_cSNV_all_df,
                                          aes(x = pre,
                                              y = corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. pre reads') +
  xlab('pre reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_cSNV_all_pre_corr_pre
```

```{r}
# r2 histogram of corr_pre counts cSNV
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = corr_pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
       axis.text.y = element_text(size = 12),
       axis.text.x = element_text(size = 12),
       strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)


```

```{r}
# r2 position vs. corr_pre SNV coverage
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)


```

```{r}
# r2 pre minus corr_pre by position
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)



# r2 pre minus corr_pre by position
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = pre - corr_pre)) +
  geom_point(aes(colour = conseq,
                 size = (1 - BRCA1regionrL42_cSNV_all_df$corr_pre / BRCA1regionrL42_cSNV_all_df$pre))) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)


```

```{r}
# r2 Number of corr_post 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_cSNV_all_corr_post_count = length(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_cSNV_all_corr_post_count
# r2 Max and min corr_post reads for each cSNV
BRCA1regionrL42_cSNV_all_min_corr_post_counts = min(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_cSNV_all_min_corr_post_counts
BRCA1regionrL42_cSNV_all_max_corr_post_counts = max(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_cSNV_all_max_corr_post_counts

# r2 post counts vs. corr_post counts:
BRCA1regionrL42_cSNV_all_post_corr_post = ggplot(BRCA1regionrL42_cSNV_all_df,
                                            aes(x = post,
                                                y = corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. post reads') +
  xlab('post reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
       strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_cSNV_all_post_corr_post
```

```{r}
# r2 histogram of corr_post counts cSNV
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = corr_post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)


```

```{r}
# r2 Position vs. corr_post SNV coverage
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)


```

```{r}
# r2 post minus corr_post by position
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)


```

```{r}
# r2 post minus corr_post by position
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) +
  geom_point(aes(colour = conseq,
                 size = (1 - corr_post / post))) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Correction (reads)') +
  xlab('Position') + theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r2 apply threshold to determined by frequency OR somehow by error correction for low counts...
# r2 looking for a dramatic drop here
ggplot(BRCA1regionrL42_cSNV_all_df,
       aes(x = pre,
           y = corr_pre / pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Fraction true SNVs') +
  xlab('Uncorrected D6 reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_abline(intercept = 1,
              slope = 0,
              lty = 2)
```
Sequencing Error Correction Analysis: Partial SNV Variants
----------------------------------
```{r}
# r2 repeat with pSNV variants (only cut end is edited!)
BRCA1region_p_hdr_snv = paste('p_hdr_', BRCA1region_cut_end,
                         '_snv', sep = '')
BRCA1regionrL42_pSNV_all <-
  which(BRCA1regionrL42[, BRCA1region_p_hdr_snv] == 'True')
BRCA1regionrL42_pSNV_all_df = BRCA1regionrL42[BRCA1regionrL42_pSNV_all,]
# r2 Number of 'pSNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_pSNV_all_lib_count = length(BRCA1regionrL42_pSNV_all_df$lib)
BRCA1regionrL42_pSNV_all_lib_count
# r2 Max and min reads for each variant
BRCA1regionrL42_pSNV_all_min_lib_counts = min(BRCA1regionrL42_pSNV_all_df$lib)
BRCA1regionrL42_pSNV_all_min_lib_counts
BRCA1regionrL42_pSNV_all_max_lib_counts = max(BRCA1regionrL42_pSNV_all_df$lib)
BRCA1regionrL42_pSNV_all_max_lib_counts
# r2 Number of pre 'SNV variants'
BRCA1regionrL42_pSNV_all_pre_count = length(BRCA1regionrL42_pSNV_all_df$pre)
BRCA1regionrL42_pSNV_all_pre_count
# r2 Max and min pre reads for each pSNV
BRCA1regionrL42_pSNV_all_min_pre_counts = min(BRCA1regionrL42_pSNV_all_df$pre)
BRCA1regionrL42_pSNV_all_min_pre_counts
BRCA1regionrL42_pSNV_all_max_pre_counts = max(BRCA1regionrL42_pSNV_all_df$pre)
BRCA1regionrL42_pSNV_all_max_pre_counts
```

```{r}
# r2 histograms of pre counts pSNV
ggplot(BRCA1regionrL42_pSNV_all_df,
       aes(x = pre, fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r2 Position vs. pre SNV coverage
ggplot(BRCA1regionrL42_pSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r2 Number of post 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_pSNV_all_post_count = length(BRCA1regionrL42_pSNV_all_df$post)
BRCA1regionrL42_pSNV_all_post_count
# r2 Max and min post reads for each pSNV
BRCA1regionrL42_pSNV_all_min_post_counts = min(BRCA1regionrL42_pSNV_all_df$post)
BRCA1regionrL42_pSNV_all_min_post_counts
BRCA1regionrL42_pSNV_all_max_post_counts = max(BRCA1regionrL42_pSNV_all_df$post)
BRCA1regionrL42_pSNV_all_max_post_counts

# r2 histograms of post counts
ggplot(BRCA1regionrL42_pSNV_all_df,
       aes(x = post,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r2 Position vs. post SNV coverage
ggplot(BRCA1regionrL42_pSNV_all_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(post))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 post reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```
Thresholded Data
----------------------------------
```{r}
# r2 Plots for thresholded data ('thresh5' indicates 1 in 1e5 or higher freq.):
# r2 Number of 'cSNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_prog_map_thresh5_lib_count = length(BRCA1regionrL42_prog_map_thresh5_df$lib)
BRCA1regionrL42_prog_map_thresh5_lib_count
BRCA1regionrL42_prog_map_thresh5_min_lib_counts = min(BRCA1regionrL42_prog_map_thresh5_df$lib)
BRCA1regionrL42_prog_map_thresh5_min_lib_counts
BRCA1regionrL42_prog_map_thresh5_max_lib_counts = max(BRCA1regionrL42_prog_map_thresh5_df$lib)
BRCA1regionrL42_prog_map_thresh5_max_lib_counts

# r2 histograms of lib counts
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = lib,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  theme(legend.position = "none") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r2 Position vs. lib SNV coverage
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(lib))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 library reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
# r2 Number of pre 'SNV variants' --> will be 2BRCA1region less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_prog_map_thresh5_pre_count = length(BRCA1regionrL42_prog_map_thresh5_df$pre)
BRCA1regionrL42_prog_map_thresh5_pre_count
BRCA1regionrL42_prog_map_thresh5_min_pre_counts = min(BRCA1regionrL42_prog_map_thresh5_df$pre)
BRCA1regionrL42_prog_map_thresh5_min_pre_counts
BRCA1regionrL42_prog_map_thresh5_max_pre_counts = max(BRCA1regionrL42_prog_map_thresh5_df$pre)
BRCA1regionrL42_prog_map_thresh5_max_pre_counts

# r2 pre counts vs. corr_pre counts:
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = pre,
           y = corr_pre)) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('corr. pre reads') +
  xlab('pre reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r2 histogram of corr_pre counts prog_map
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = corr_pre,
           fill = conseq,
           col = conseq)) +
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Single Nucleotide Variants') +
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r2 Position vs. pre SNV coverage
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y =  log2(pre))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Log-2 pre reads') +
  xlab('Position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

# r2 Position vs. corr_pre SNV coverage
ggplot(BRCA1regionrL42_prog_map_thresh5_df, 
       aes(x = as.numeric(as.character(pos)), 
           y = log2(corr_pre))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 pre reads') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 pre minus corr_pre by position
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)), 
           y = pre - corr_pre)) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Correction (reads)') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 pre minus corr_pre by position with size
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)), 
           y = pre - corr_pre)) + 
  geom_point(aes(colour = conseq,
         size = (1 - BRCA1regionrL42_prog_map_thresh5_df$corr_pre / BRCA1regionrL42_prog_map_thresh5_df$pre))) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Correction (reads)') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)
```

```{r}
# r2 Number of post 'SNV variants' --> will be 2 less than SGE region length x 3 due to PAM edits
BRCA1regionrL42_prog_map_thresh5_post_count = length(BRCA1regionrL42_prog_map_thresh5_df$post)
BRCA1regionrL42_prog_map_thresh5_post_count
BRCA1regionrL42_prog_map_thresh5_min_post_counts = min(BRCA1regionrL42_prog_map_thresh5_df$post)
BRCA1regionrL42_prog_map_thresh5_min_post_counts
BRCA1regionrL42_prog_map_thresh5_max_post_counts = max(BRCA1regionrL42_prog_map_thresh5_df$post)
BRCA1regionrL42_prog_map_thresh5_max_post_counts
BRCA1regionrL42_prog_map_thresh5_corr_post_count = length(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_prog_map_thresh5_corr_post_count
BRCA1regionrL42_prog_map_thresh5_min_corr_post_counts = min(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_prog_map_thresh5_min_corr_post_counts
BRCA1regionrL42_prog_map_thresh5_max_corr_post_counts = max(BRCA1regionrL42_cSNV_all_df$corr_post)
BRCA1regionrL42_prog_map_thresh5_max_corr_post_counts
```

```{r}
# r2 post counts vs. corr_post counts:
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = post, 
           y = corr_post)) +
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('corr. post reads') +
  xlab('post reads') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

# r2 histograms of post counts
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = post, 
           fill = conseq, 
           col = conseq)) + 
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Single Nucleotide Variants') +
  xlab('Reads') + 
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

# r2 histogram of corr_post counts prog_map
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = corr_post, 
           fill = conseq, 
           col = conseq)) + 
  geom_histogram(bins = 50) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Single Nucleotide Variants') +
  xlab('Reads') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme( text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

# r2 Position vs. post SNV coverage
ggplot(BRCA1regionrL42_prog_map_thresh5_df, 
       aes(x = as.numeric(as.character(pos)), 
           y = log2(post))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 post reads') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 Position vs. corr_post SNV coverage
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = log2(corr_post))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 post reads') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 post minus corr_post by position
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)),
           y = post - corr_post)) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Correction (reads)') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 post minus corr_post by position
ggplot(BRCA1regionrL42_prog_map_thresh5_df,
       aes(x = as.numeric(as.character(pos)), 
           y = post - corr_post)) + 
  geom_point(aes(colour = conseq,
                 size = (1 - BRCA1regionrL42_prog_map_thresh5_df$corr_post / BRCA1regionrL42_prog_map_thresh5_df$post))) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Correction (reads)') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

# r2 looking for a dramatic drop here
ggplot(BRCA1regionrL42_prog_map_thresh5_df, 
       aes(x = pre, 
           y = corr_pre / pre)) +
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('corr_pre/pre') +
  xlab('pre reads') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_abline(intercept = 1, 
              slope = 0,
              lty = 2)

BRCA1regionrL42_pSNV_thresh5 <-
  which(BRCA1regionrL42[, BRCA1region_p_hdr_snv] == 'True' & as.numeric(as.character(BRCA1regionrL42[, 'pre_freq'])) >= 0.00001)
BRCA1regionrL42_pSNV_thresh5_df = BRCA1regionrL42[BRCA1regionrL42_pSNV_thresh5, ]
BRCA1regionrL42_pSNV_thresh5_lib_count = length(BRCA1regionrL42_pSNV_thresh5_df$lib)
BRCA1regionrL42_pSNV_thresh5_lib_count
BRCA1regionrL42_pSNV_thresh5_min_lib_counts = min(BRCA1regionrL42_pSNV_thresh5_df$lib)
BRCA1regionrL42_pSNV_thresh5_min_lib_counts
BRCA1regionrL42_pSNV_thresh5_max_lib_counts = max(BRCA1regionrL42_pSNV_thresh5_df$lib)
BRCA1regionrL42_pSNV_thresh5_max_lib_counts
BRCA1regionrL42_pSNV_thresh5_pre_count = length(BRCA1regionrL42_pSNV_thresh5_df$pre)
BRCA1regionrL42_pSNV_thresh5_pre_count
BRCA1regionrL42_pSNV_thresh5_min_pre_counts = min(BRCA1regionrL42_pSNV_thresh5_df$pre)
BRCA1regionrL42_pSNV_thresh5_min_pre_counts
BRCA1regionrL42_pSNV_thresh5_max_pre_counts = max(BRCA1regionrL42_pSNV_thresh5_df$pre)
BRCA1regionrL42_pSNV_thresh5_max_pre_counts
```

```{r}
# r2 histograms of pre counts pSNV
ggplot(BRCA1regionrL42_pSNV_thresh5_df,
       aes(x = pre, fill = conseq, 
           col = conseq)) + 
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Single Nucleotide Variants') +
  xlab('Reads') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

# r2 Position vs. pre pSNV coverage
ggplot(BRCA1regionrL42_pSNV_thresh5_df, 
       aes(x = as.numeric(as.character(pos)), 
           y = log2(pre))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 pre reads') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.text.x = element_text(size = 12),
    strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)

BRCA1regionrL42_pSNV_thresh5_post_count = length(BRCA1regionrL42_pSNV_thresh5_df$post)
BRCA1regionrL42_pSNV_thresh5_post_count
BRCA1regionrL42_pSNV_thresh5_min_post_counts = min(BRCA1regionrL42_pSNV_thresh5_df$post)
BRCA1regionrL42_pSNV_thresh5_min_post_counts
BRCA1regionrL42_pSNV_thresh5_max_post_counts = max(BRCA1regionrL42_pSNV_thresh5_df$post)
BRCA1regionrL42_pSNV_thresh5_max_post_counts
```

```{r}
# r2 histograms of post counts
ggplot(BRCA1regionrL42_pSNV_thresh5_df,
       aes(x = post,
           fill = conseq,
           col = conseq)) + 
  geom_histogram(bins = 50) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Single Nucleotide Variants') +
  xlab('Reads') + 
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)
```

```{r}
# r2 Position vs. post SNV coverage
ggplot(BRCA1regionrL42_pSNV_thresh5_df, 
       aes(x = as.numeric(as.character(pos)),
           y = log2(post))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 post reads') +
  xlab('Position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, 
             lty = 2)
```
Replicate Analysis
===================
```{r}
#define intersection based on most broad category that you care about, and then filter the others based on that intersection:
#Intersection variants of BRCA1regionrL41_prog_var_thresh5_df:
BRCA1regionrL41_prog_var_thresh5_df_rL42_int <-
  BRCA1regionrL41_prog_var_thresh5_df[which(
    BRCA1regionrL41_prog_var_thresh5_df$variant %in% BRCA1regionrL42_prog_var_thresh5_df$variant),]
#Intersection variants of BRCA1regionrL42_prog_var_thresh5_df:
BRCA1regionrL42_prog_var_thresh5_df_rL41_int <-
  BRCA1regionrL42_prog_var_thresh5_df[which(
    BRCA1regionrL42_prog_var_thresh5_df$variant %in% BRCA1regionrL41_prog_var_thresh5_df$variant), ]
#Order each one by variant (check to see if same):
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int[order(BRCA1regionrL41_prog_var_thresh5_df_rL42_int$variant), ]
BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int[order(BRCA1regionrL42_prog_var_thresh5_df_rL41_int$variant), ]

dim(BRCA1regionrL41_prog_var_thresh5_df)[1] #Number of prog_vars in rL41
BRCA1regionrL41rL42int_prog_var_thresh5_pre_count <-
  dim(BRCA1regionrL41_prog_var_thresh5_df_rL42_int)[1]
BRCA1regionrL41rL42int_prog_var_thresh5_pre_count #Number of prog_vars in rL41 also in rL42
dim(BRCA1regionrL42_prog_var_thresh5_df)[1] #Number of prog_vars in rL42
dim(BRCA1regionrL42_prog_var_thresh5_df_rL41_int)[1] #Number of prog_vars in rL42 also in rL41

#Add the following columns from the from rL42 df to the rL41 df:
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_lib <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$lib
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre_pseudo
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pseudo <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_pseudo
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna_pseudo <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna_pseudo
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_lib_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre_lib_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pre_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_pre_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_lib_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_lib_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna_pre_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna_pre_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_rna_post_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$rna_post_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_snv_error_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$pre_snv_error_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_snv_error_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$post_snv_error_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_exp_snv_error_reads_pre <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$exp_snv_error_reads_pre
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_exp_snv_error_reads_post <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$exp_snv_error_reads_post
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_corr_pre <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$corr_pre
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_corr_post <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$corr_post
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_corr_pre_pseudo <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$corr_pre_pseudo
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_corr_post_pseudo <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$corr_post_pseudo
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_corr_pseudo_post_pre_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$corr_pseudo_post_pre_ratio

# tHDR entries pulled over to match...
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_pre
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_lib <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_lib
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_post
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_rna
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_pre_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_post_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pseudo_freq <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_rna_pseudo_freq
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_lib_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_pre_lib_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pre_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_post_pre_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_lib_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_post_lib_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pre_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_rna_pre_ratio
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_post_ratio <-
  BRCA1regionrL42_prog_var_thresh5_df_rL41_int_ord$tHDR_rna_post_ratio

#compute new columns
#pre_lib_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_lib_ratio) / 2
#post_pre_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pre_ratio) / 2
#post_lib_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_lib_ratio) / 2
#tHDR_pre_lib_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_lib_ratio) / 2
#tHDR_post_pre_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pre_rati) / 2
#tHDR_post_lib_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_lib_ratio) / 2
#tHDR_rna_pre_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pre_ratio) / 2
#tHDR_rna_post_ratio_rL41rL42_mean
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio_rL41rL42_mean <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_post_ratio) / 2

#weighting by numerator:  <- ((a*a/(a+c))+(c*c/(a+c)))/b
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_rL41rL42_pre_weighted <-
  ((BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq * BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq / (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq)) +
     (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq * BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq / (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq))) / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$lib_pseudo_freq

#rL41 pre-freq weighting coefficient (rL41_pre-freq / (rL41 + rL42 pre-freq's))
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_weight <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq / (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq)
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_weight <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq / (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq)

#this will be weighted by denominator:  (rL41_pof/rL41_prf)(rL41_prwc)+(rL42_pof/rL42_prf)*(rL42_prwc)
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_pre_weighted <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pseudo_freq / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_weight) + (
    BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pseudo_freq / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_weight)

#this will be weighted by denominator:  (rL41_pof/lif')*(rL41_prwc)+(rL42_pof/lif)*(rL42_prwc)
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_rL41rL42_pre_weighted <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pseudo_freq / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$lib_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_weight) + (
    BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pseudo_freq / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$lib_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_weight)

#rL41 tHDR_pre-freq weighting coefficient (rL41_tHDR_pre-freq / (rL41 + rL42 pre-freq's))
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_weight <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_pseudo_freq / (
    BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_pseudo_freq)
#rL42 tHDR_pre-freq weighting coefficient
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_weight <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_pseudo_freq /
  (
    BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_pseudo_freq + BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_pseudo_freq)
#this will be weighted by denominator:  (rL41_pof/rL41_prf)(rL41_prwc)+(rL42_pof/rL42_prf)*(rL42_prwc)
#tHDR_post_pre_ratio_rL41rL42_pre_weighted
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_pre_weighted <-
  (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pseudo_freq / BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_weight) + (
    BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pseudo_freq /
      BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_pseudo_freq
  ) * (BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_weight)
```

Calculate quartiles and distributions for each functional class of variant:
```{r}

```
Create syn-normalized (synnorm) values -- modified for BRCA1region,U1,U2 to be median of experiment
```{r}
#### COPIED LIST OF CATEGORIES NOW IN DF -  pick which ones to normalize to median synonymous variant:
#rL41 #leave 'synnorm' in place even though it's normalized to median of all SNVs
BRCA1regionrL41_cSNV_pre_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio)
BRCA1regionrL41_cSNV_post_pre_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio)
BRCA1regionrL41_cSNV_post_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio)

BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio / BRCA1regionrL41_cSNV_pre_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio / BRCA1regionrL41_cSNV_post_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio / BRCA1regionrL41_cSNV_post_lib_med
# tHDR
BRCA1regionrL41_tSNV_pre_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio)
BRCA1regionrL41_tSNV_post_pre_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio)
BRCA1regionrL41_tSNV_post_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio)
BRCA1regionrL41_tSNV_rna_pre_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio)
BRCA1regionrL41_tSNV_rna_post_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio)

BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio / BRCA1regionrL41_tSNV_pre_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio / BRCA1regionrL41_tSNV_post_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio / BRCA1regionrL41_tSNV_post_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio / BRCA1regionrL41_tSNV_rna_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio / BRCA1regionrL41_tSNV_rna_post_med

#rL42
BRCA1regionrL42_cSNV_pre_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_lib_ratio)
BRCA1regionrL42_cSNV_post_pre_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pre_ratio)
BRCA1regionrL42_cSNV_post_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_lib_ratio)

BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_pre_lib_ratio / BRCA1regionrL42_cSNV_pre_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pre_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_pre_ratio / BRCA1regionrL42_cSNV_post_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_lib_ratio_synnorm <-
  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_post_lib_ratio / BRCA1regionrL42_cSNV_post_lib_med
# tHDR
BRCA1regionrL42_tSNV_pre_lib_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_lib_ratio)
BRCA1regionrL42_tSNV_post_pre_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pre_ratio)
BRCA1regionrL42_tSNV_post_lib_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_lib_ratio)
BRCA1regionrL42_tSNV_rna_pre_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pre_ratio)
BRCA1regionrL42_tSNV_rna_post_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_post_ratio)

BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_lib_ratio_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_pre_lib_ratio/BRCA1regionrL42_tSNV_pre_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pre_ratio_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_pre_ratio/BRCA1regionrL42_tSNV_post_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_lib_ratio_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_post_lib_ratio/BRCA1regionrL42_tSNV_post_lib_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pre_ratio_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_pre_ratio/BRCA1regionrL42_tSNV_rna_pre_med
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_post_ratio_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$rL42_tHDR_rna_post_ratio/BRCA1regionrL42_tSNV_rna_post_med

###normalize the means next!!!
BRCA1regionrL41rL42a_cSNV_pre_lib_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_cSNV_post_pre_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_cSNV_post_lib_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_tSNV_pre_lib_med <- 
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_tSNV_post_pre_med <- median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_tSNV_post_lib_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_tSNV_rna_pre_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio_rL41rL42_mean)
BRCA1regionrL41rL42a_tSNV_rna_post_med <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio_rL41rL42_mean)
BRCA1regionrL41rL42pw_tSNV_post_pre_med  <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_pre_weighted)
BRCA1regionrL41rL42pw_cSNV_post_pre_med  <-
  median(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_pre_weighted)

#pre_lib_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$pre_lib_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_cSNV_pre_lib_med
#post_pre_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_cSNV_post_pre_med
#post_lib_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_lib_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_cSNV_post_lib_med
#tHDR_pre_lib_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_pre_lib_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_tSNV_pre_lib_med
#tHDR_post_pre_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_tSNV_post_pre_med
#tHDR_post_lib_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_lib_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_tSNV_post_lib_med
#tHDR_rna_pre_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_pre_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_tSNV_rna_pre_med
#tHDR_rna_post_ratio_rL41rL42_mean_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio_rL41rL42_mean_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_rna_post_ratio_rL41rL42_mean/BRCA1regionrL41rL42a_tSNV_rna_post_med
#tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$tHDR_post_pre_ratio_rL41rL42_pre_weighted/BRCA1regionrL41rL42pw_tSNV_post_pre_med
#post_pre_ratio_rL41rL42_pre_weighted_synnorm
BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_pre_weighted_synnorm <- BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord$post_pre_ratio_rL41rL42_pre_weighted/BRCA1regionrL41rL42pw_cSNV_post_pre_med

#working intersection dictionary:  BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord = which(BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord[,'pos']!='REF')
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df = BRCA1regionrL41_prog_var_thresh5_df_rL42_int_ord[BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord,]
BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord = which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[,'cDNApos']!='REF' & BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[,'cDNApos']!='NA')
BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord,]
BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord = which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[,'CADD.phred']!='NA')
BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord,]

BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense <- BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df[which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$conseq=='NON_SYNONYMOUS'),]
```
Replciate Plotting
------------------
```{r}
###Replicate Plotting
#Plot comparing post_pre_ratio from each experiment:
#BRCA1regionrL41rL42_prog_map_rL41_post_pre_by_rL42_post_pre_alpha <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_synnorm),
           y = log2(rL42_post_pre_ratio_synnorm))) + 
  geom_point(aes(colour = conseq,
                 alpha = (pre_freq + rL42_pre_freq)),
             size = 1.5) + 
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) + 
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) + 
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('Rep. 2 enrichment') +
  xlab('Rep. 1 enrichment') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12), 
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
#BRCA1regionrL41rL42_prog_map_rL41_post_pre_by_rL42_post_pre <-
BRCA1regionrL4_post_pre_ratio_across_reps = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_synnorm),
           y = log2(rL42_post_pre_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  xlim(-4, 1) +
  ylim(-4, 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) + 
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 enrichment') +
  xlab('Rep. 1 enrichment') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL4_post_pre_ratio_across_reps
```

```{r}
#correlation between intersection of replicates, log-2 normalized:
BRCA1regionrL41rL42_cSNV_rL41_post_pre_by_rL42_post_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_pre_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_pre_ratio_synnorm),
    method = 'spearman')
BRCA1regionrL41rL42_cSNV_rL41_post_pre_by_rL42_post_pre_cor_s

BRCA1regionrL41rL42_cSNV_rL41_post_pre_by_rL42_post_pre_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_pre_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_pre_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_cSNV_rL41_post_pre_by_rL42_post_pre_cor_p
```

```{r}
#Plot comparing post_lib_ratio from each experiment with colors for mutation consequence:
#BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_alpha <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_lib_ratio_synnorm),
           y = log2(rL42_post_lib_ratio_synnorm))) +
  geom_point(aes(colour = conseq,
                 alpha = (pre_freq + rL42_pre_freq)),
             size = 1.5) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) + 
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('Rep. 2 enrichment') + 
  xlab('Rep. 1 enrichment') + 
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
#Same plot comparing post_pseudo_freq/lib_pseudo_freq from each experiment, no alpha:
#BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_lib_ratio_synnorm),
           y = log2(rL42_post_lib_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1.5) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 enrichment') +
  xlab('Rep. 1 enrichment') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#correlation between intersection of replicates, log-2 normalized:
BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_lib_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_lib_ratio_synnorm),
    method = 'spearman')
BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_cor_s

BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_lib_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_lib_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_cor_p
```

```{r}
#Plot comparing tHDR_post_pre_ratio from each experiment:
#BRCA1regionrL41rL42_prog_map_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre_alpha <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_pre_ratio_synnorm),
           y = log2(rL42_tHDR_post_pre_ratio_synnorm))) +
  geom_point(aes(colour = conseq,
                 alpha = (pre_freq + rL42_pre_freq)),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 enrichment') +
  xlab('Rep. 1 enrichment') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
```

```{r}
#BRCA1regionrL41rL42_prog_map_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre <-
BRCA1regionrL4_tHDR_post_pre_across_reps = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_pre_ratio_synnorm),
           y = log2(rL42_tHDR_post_pre_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1.5) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  xlim(-1,1) +
  ylim(-1,1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 enrichment') +
  xlab('Rep. 1 enrichment') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL4_tHDR_post_pre_across_reps
```

```{r}
#correlation between intersection of replicates, log-2 normalized:
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm),
    method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre_cor_s

BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_pre_by_rL42_tHDR_post_pre_cor_p
```

```{r}
#Plot comparing post_lib_ratio from each experiment with colors for mutation consequence:
#BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib_alpha <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_lib_ratio_synnorm),
           y = log2(rL42_tHDR_post_lib_ratio_synnorm))) +
  geom_point(aes(colour = conseq,
                 alpha = (pre_freq + rL42_pre_freq)),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 post/lib') +
  xlab('Rep. 1 post/lib') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)
#Same plot comparing post_pseudo_freq/lib_pseudo_freq from each experiment, no alpha:
#BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib <-
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_lib_ratio_synnorm),
           y = log2(rL42_tHDR_post_lib_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) + theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),legend.key = element_blank()) +
  ylab('Rep. 2 post/lib') +
  xlab('Rep. 1 post/lib') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#correlation between intersection of replicates, log-2 normalized:
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_lib_ratio_synnorm),
    method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib_cor_s

BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_lib_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_post_lib_by_rL42_tHDR_post_lib_cor_p
```

```{r}
#Plot comparing rna_pre_ratio from each experiment with colors for mutation consequence:
#BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre_alpha <-
ggplot(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_rna_pre_ratio_synnorm),
           y = log2(rL42_tHDR_rna_pre_ratio_synnorm))) +
  geom_point(aes(colour = conseq,
                 alpha = (pre_freq + rL42_pre_freq)),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 rna/pre') +
  xlab('Rep. 1 rna/pre') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#Same plot comparing post_pseudo_freq/lib_pseudo_freq from each experiment, no alpha:
#BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre <-
ggplot(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_rna_pre_ratio_synnorm), 
           y = log2(rL42_tHDR_rna_pre_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Rep. 2 rna/pre') +
  xlab('Rep. 1 rna/pre') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#correlation between intersection of replicates, log-2 normalized:
BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$tHDR_rna_pre_ratio_synnorm), 
      log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rL42_tHDR_rna_pre_ratio_synnorm),
    method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre_cor_s

BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$tHDR_rna_pre_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rL42_tHDR_rna_pre_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_tSNV_rL41_tHDR_rna_pre_by_rL42_tHDR_rna_pre_cor_p
```

```{r}
#Plot comparing pre_freq reads from each experiment
BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre <- ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
                                                   aes(x = log10(pre_freq),
                                                       y = log10(rL42_pre_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Replicate 2 pre reads (Log-10)') +
  xlab('Replicate 1 pre reads (Log-10)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 20),
        axis.text.y = element_text(size = 20),
        axis.text.x = element_text(size = 20),
        strip.background = element_blank(),
        legend.position = "none") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre
```

```{r}
#correlations between replicates, pre_freq reads:
BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_freq),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_freq),
      method = 'spearman')
BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre_cor_s

BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_freq),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_freq),
      method = 'pearson')
BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre_cor_p
```

```{r}
#Plot comparing post_freq reads from each experiment
#BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post <-
BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pseudo_freq),
           y = log2(rL42_post_pseudo_freq))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title=element_blank(),
        legend.key = element_blank()) +
  ylab('Replicate 2 post reads, corrected (Log-2)') +
  xlab('Replicate 1 post reads, corrected (Log-2)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank(),
        legend.position = "none") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post
```

```{r}
#correlations between replicates, post_freq reads:
BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_freq),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_freq),
      method = 'spearman')
BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post_cor_s

BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_pseudo_freq),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_pseudo_freq),
      method = 'pearson')
BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post_cor_p
```

```{r}
#Plot comparing pre_lib_ratio across replicates
#BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib <- 
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(pre_lib_ratio_synnorm),
           y = log2(rL42_pre_lib_ratio_synnorm))) +
  geom_point(aes(colour = conseq),
             size = 1) +
  geom_abline(intercept = 0,
              slope = 1,
              lty = 2) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Replicate 2 pre/lib (Log-2)') +
  xlab('Replicate 1 pre/lib (Log-2)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank(),
        legend.position = "none") +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#correlations between replicates, pre_lib_ratio:
BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_lib_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_lib_ratio_synnorm),
      method = 'spearman')
BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s

BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib_cor_p <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_lib_ratio_synnorm),
    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_lib_ratio_synnorm),
    method = 'pearson')
BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib_cor_p
```

```{r}
# pre_lib_ratio (average across replicates) by position
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pre_pos_effect_intercept = coef(lm(log2(pre_lib_ratio_rL41rL42_mean_synnorm) ~ pos,
  data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[1]

BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pre_pos_effect_slope = coef(lm(log2(pre_lib_ratio_rL41rL42_mean_synnorm) ~ pos, 
  data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[2]
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pre_pos_effect_slope #decrease in editing efficiency per bp

ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(pre_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Pre-freq / lib-freq (Log-2, ave. of reps)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_abline(intercept = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pre_pos_effect_intercept,
              slope = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pre_pos_effect_slope,
              lty = 1) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
#tHDR_pre_lib_ratio by pos using mean:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq), 
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title =
          element_blank(),
        legend.key = element_blank()) +
  ylab('Enrichment (pre/lib mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL41rL42_tSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_pre_lib_ratio_synnorm),
      method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s
```

```{r}
#tHDR_post_lib_ratio by pos using mean:
BRCA1regionrL4_tHDR_post_lib_ratio = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('Enrichment (post/lib mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL4_tHDR_post_lib_ratio
```

```{r}
BRCA1regionrL41rL42_tSNV_rL41_post_lib_by_rL42_post_lib_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_lib_ratio_synnorm),
      method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_post_lib_by_rL42_post_lib_cor_s
```

```{r}
#tHDR_post_pre_ratio by pos using mean:
BRCA1regionrL4_tHDR_post_pre_ratio = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
                                       aes(y = log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                                           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.5) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Enrichment (post/pre mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 20),
        axis.text.y = element_text(size = 20),
        axis.text.x = element_text(size = 20),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL4_tHDR_post_pre_ratio
```

```{r}
BRCA1regionrL41rL42_tSNV_rL41_post_pre_by_rL42_post_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm),
      method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_post_pre_by_rL42_post_pre_cor_s
```

```{r}
#tHDR_post_pre_ratio by pos using mean:
BRCA1regionrL4_tHDR_post_pre_ratio_pos = ggplot(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df,
                                       aes(y = log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),
                                           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Enrichment (post/pre mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size=12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL4_tHDR_post_pre_ratio_pos
```

```{r}
BRCA1regionrL41rL42_tSNV_rL41_rna_pre_by_rL42_rna_pre_cor_s <-
  cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_rna_pre_ratio_synnorm),
      log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_rna_pre_ratio_synnorm),
      method = 'spearman')
BRCA1regionrL41rL42_tSNV_rL41_rna_pre_by_rL42_rna_pre_cor_s
```

```{r}
#post_lib_ratio (mean_synnorm across reps) by position
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_post_pos_effect_intercept = coef(lm(log2(post_lib_ratio_rL41rL42_mean_synnorm) ~ pos, data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[1]
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_post_pos_effect_slope = coef(lm(log2(post_lib_ratio_rL41rL42_mean_synnorm) ~ pos,
 data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[2]
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_post_pos_effect_slope #decrease in editing efficiency per bp

BRCA1regionrL4_post_lib_freq_by_pos = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha=1,
             size = 1) +
  geom_abline(intercept = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_post_pos_effect_intercept, 
              slope=BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_post_pos_effect_slope ) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Post-freq / lib-freq (Log-2, ave. of reps)') +
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) + 
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL4_post_lib_freq_by_pos
```

```{r}
#post_pre_ratio_rL41rL42_mean_synnorm by position
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pp_pos_effect_intercept = coef(lm(log2(post_pre_ratio_rL41rL42_mean_synnorm) ~ pos,
  data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[1]
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pp_pos_effect_slope = coef(lm(log2(post_pre_ratio_rL41rL42_mean_synnorm) ~ pos, 
  data = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df))[2]
BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pp_pos_effect_slope #decrease in editing efficiency per bp

BRCA1regionrL4_post_pre_freq_by_pos = ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 2) +
  geom_abline(intercept = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pp_pos_effect_intercept,
              slope = BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df_pp_pos_effect_slope ) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  ylab('Post-freq / pre-freq (Log-2, ave. of reps)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)

BRCA1regionrL4_post_pre_freq_by_pos
```

```{r}
#pre_lib_ratio by pos using pre weighted averages:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(pre_lib_ratio_rL41rL42_pre_weighted),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('Enrichment (D20/lib, corr., weighted mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
#post_pre_ratio by pos using weighted averages:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('post/pre, log-2 weighted mean)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
#tHDR_post_pre_ratio by pos using weighted averages:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title=element_blank(),
        legend.key = element_blank()) +
  ylab('Enrichment (D20/D6, corr., weighted mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```

```{r}
#Post / lib ratio by pos using pre weighted averages:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_lib_ratio_rL41rL42_pre_weighted), 
           x = as.numeric(as.character(pos)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + 
  ylab('Enrichment (D20/lib, corr., weighted mean, log-2)') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors) +
  geom_vline(xintercept = BRCA1region_cut_pos,
             lty = 2)
```
Comparisons to CADD
===================
```{r}
#single replicate comparisons for post/lib and post/pre (only on variants passing thresh in both reps)
#BRCA1regionrL41_cSNV_post_pre_ratio_synnorm_by_CADD <- 
BRCA1regionrL41_cSNV_post_pre_ratio_synnorm_by_CADD = ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) + xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_cSNV_post_pre_ratio_synnorm_by_CADD
```

```{r}
#calculate post_pre_ratio_synnorm CADD correlation using rL41:
BRCA1regionrL41_cSNV_post_pre_by_cadd_cor_s = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_pre_ratio_synnorm),
as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
method = 'spearman')
BRCA1regionrL41_cSNV_post_pre_by_cadd_cor_s
```

```{r}
#BRCA1regionrL42_cSNV_post_pre_ratio_synnorm_by_CADD <-
BRCA1regionrL42_cSNV_post_pre_ratio_synnorm_by_CADD = ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(rL42_post_pre_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_cSNV_post_pre_ratio_synnorm_by_CADD
```

```{r}
#calculate post_pre_ratio_synnorm CADD correlation using rL42:
BRCA1regionrL42_cSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$rL42_post_pre_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL42_cSNV_post_pre_by_cadd_cor_s
```

```{r}
#BRCA1regionrL41_cSNV_post_lib_ratio_synnorm_by_CADD <- 
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_lib_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors) +
  scale_color_manual(values = named_conseq_colors)

#calculate post_lib_ratio_synnorm CADD correlation using rL41:
BRCA1regionrL41_cSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_lib_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41_cSNV_post_lib_by_cadd_cor_s
```

```{r}
#BRCA1regionrL42_cSNV_post_lib_ratio_synnorm_by_CADD <- 
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(rL42_post_lib_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate post_lib_ratio_synnorm CADD correlation using rL42:
BRCA1regionrL42_cSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$rL42_post_lib_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL42_cSNV_post_lib_by_cadd_cor_s
```

```{r}
#calculate individual replicate correlations to CADD with tHDR (i.e. tSNV)
BRCA1regionrL41_tSNV_tHDR_post_pre_ratio_synnorm_by_CADD = ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_pre_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL41_tSNV_tHDR_post_pre_ratio_synnorm_by_CADD
```

```{r}
#calculate tHDR_post_pre_ratio_synnorm CADD correlation using rL41:
BRCA1regionrL41_tSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41_tSNV_post_pre_by_cadd_cor_s
```

```{r}
#BRCA1regionrL42_tSNV_tHDR_post_pre_ratio_synnorm_by_CADD <-
BRCA1regionrL42_tSNV_tHDR_post_pre_ratio_synnorm_by_CADD = ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(rL42_tHDR_post_pre_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL42_tSNV_tHDR_post_pre_ratio_synnorm_by_CADD
```

```{r}
#calculate tHDR_post_pre_ratio_synnorm CADD correlation using rL42:
BRCA1regionrL42_tSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL42_tSNV_post_pre_by_cadd_cor_s
```

```{r}
#BRCA1regionrL41_tSNV_tHDR_post_lib_ratio_synnorm_by_CADD <- 
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_lib_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate tHDR_post_lib_ratio_synnorm CADD correlation using rL41:
BRCA1regionrL41_tSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41_tSNV_post_lib_by_cadd_cor_s
```

```{r}
#BRCA1regionrL42_tSNV_tHDR_post_lib_ratio_synnorm_by_CADD <- 
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(rL42_tHDR_post_lib_ratio_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate tHDR_post_lib_ratio_synnorm CADD correlation using rL42:
BRCA1regionrL42_tSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_lib_ratio_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL42_tSNV_post_lib_by_cadd_cor_s
```

```{r}
#calculate weighted average correlations with CADD for cSNV
#Average pre_lib_ratio plotted vs. cadd
BRCA1regionrL4_pre_lib_ratio_vs_cadd = ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(pre_lib_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D6/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

BRCA1regionrL4_pre_lib_ratio_vs_cadd
```

```{r}
#calculate CADD correlation using average:
BRCA1regionrL41rL42a_cSNV_pre_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$pre_lib_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_cSNV_pre_lib_by_cadd_cor_s
```

```{r}
#Weighted pre_lib_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(pre_lib_ratio_rL41rL42_pre_weighted),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D6/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using weighted averages:
BRCA1regionrL41rL42w_cSNV_pre_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$pre_lib_ratio_rL41rL42_pre_weighted),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42w_cSNV_pre_lib_by_cadd_cor_s
```

```{r}
#Average post_pre_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using average:
BRCA1regionrL41rL42a_cSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_pre_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_cSNV_post_pre_by_cadd_cor_s
```

```{r}
#Weighted post_pre_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           y = as.numeric(as.character(CADD.phred)))) +
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using weighted averages:
BRCA1regionrL41rL42w_cSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_pre_ratio_rL41rL42_pre_weighted_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42w_cSNV_post_pre_by_cadd_cor_s
```

```{r}
#Average post_lib_ratio_synnorm plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_lib_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) +
  xlab('Enrichment (D20/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using average:
BRCA1regionrL41rL42a_cSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_lib_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_cSNV_post_lib_by_cadd_cor_s
```

```{r}
#Weighted post / lib ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_lib_ratio_rL41rL42_pre_weighted),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
    theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using weighted averages:
BRCA1regionrL41rL42w_cSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_lib_ratio_rL41rL42_pre_weighted),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42w_cSNV_post_lib_by_cadd_cor_s
```

```{r}
#calculate average correlations with CADD for tSNV
#Average tHDR_pre_lib_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D6/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
    theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using average:
BRCA1regionrL41rL42a_tSNV_pre_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_tSNV_pre_lib_by_cadd_cor_s
```

```{r}
#Average tHDR_post_pre_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
    theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using average:
BRCA1regionrL41rL42a_tSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_tSNV_post_pre_by_cadd_cor_s
```

```{r}
#Weighted tHDR_post_pre_ratio plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/D6, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
    theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using weighted averages:
BRCA1regionrL41rL42w_tSNV_post_pre_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42w_tSNV_post_pre_by_cadd_cor_s
```

```{r}
#Average tHDR_post_lib_ratio_synnorm plotted vs. cadd
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  xlab('Enrichment (D20/lib, mean, log-2)') +
  ylab('CADD Score (Phred)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)

#calculate CADD correlation using average:
BRCA1regionrL41rL42a_tSNV_post_lib_by_cadd_cor_s = cor(
  log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
  as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
  method = 'spearman')
BRCA1regionrL41rL42a_tSNV_post_lib_by_cadd_cor_s
```

```{r}
#post_pre_ratio_rL41rL42_pre_weighted_synnorm density plots for all SNVs
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           fill = conseq)) + 
  geom_density(alpha = 0.5) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('SNV density') +
  xlab('Enrichment (Log-2)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)
```

```{r}
#post_pre_ratio_rL41rL42_pre_weighted_synnorm density plot for only missynstop
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df[which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$conseq == '5PRIME_UTR'), ],
       aes(x = log2(post_pre_ratio_rL41rL42_pre_weighted_synnorm),
           fill = conseq)) + 
  geom_density(alpha = 0.5) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('SNV density') +
  xlab('Enrichment (Log-2)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)
```

```{r}
#post_pre_ratio_rL41rL42_mean_synnorm density plots based on mean
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           fill = conseq)) + 
  geom_density(alpha = 0.5) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('SNV density') +
  xlab('Enrichment (Log-2)') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors)
```
Summary Data
============
```{r}
#variant coverage metrics:
BRCA1regionrL41_prog_var_all_lib_count
BRCA1regionrL41_prog_var_all_pre_count
BRCA1regionrL42_prog_var_all_pre_count
BRCA1regionrL41_prog_map_thresh5_lib_count
BRCA1regionrL41_prog_map_thresh5_pre_count
BRCA1regionrL42_prog_map_thresh5_pre_count
BRCA1regionrL41_prog_map_thresh5_post_count
BRCA1regionrL42_prog_map_thresh5_post_count
BRCA1regionrL41rL42int_prog_var_thresh5_pre_count
#correlations between replicate frequencies (all psuedocounted, spearman rank)
BRCA1regionrL41rL42_cSNV_rL41_pre_by_rL42_pre_cor_s
BRCA1regionrL41rL42_cSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s
BRCA1regionrL41rL42_cSNV_rL41_post_by_rL42_post_cor_s
BRCA1regionrL41rL42_cSNV_rL41_post_lib_by_rL42_post_lib_cor_s
BRCA1regionrL41rL42_cSNV_rL41_post_pre_by_rL42_post_pre_cor_s
#add tSNV correlations...
BRCA1regionrL41rL42_tSNV_rL41_post_pre_by_rL42_post_pre_cor_s
BRCA1regionrL41rL42_tSNV_rL41_pre_lib_by_rL42_pre_lib_cor_s
BRCA1regionrL41rL42_tSNV_rL41_post_lib_by_rL42_post_lib_cor_s
BRCA1regionrL41rL42_tSNV_rL41_rna_pre_by_rL42_rna_pre_cor_s

#correlations with CADD for single replicates
BRCA1regionrL41_cSNV_post_pre_by_cadd_cor_s
BRCA1regionrL42_cSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41_cSNV_post_lib_by_cadd_cor_s
BRCA1regionrL42_cSNV_post_lib_by_cadd_cor_s
BRCA1regionrL41_tSNV_post_pre_by_cadd_cor_s
BRCA1regionrL42_tSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41_tSNV_post_lib_by_cadd_cor_s
BRCA1regionrL42_tSNV_post_lib_by_cadd_cor_s
#correlations with CADD for weighted or non-weighted averages
BRCA1regionrL41rL42a_cSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41rL42w_cSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41rL42a_cSNV_post_lib_by_cadd_cor_s
BRCA1regionrL41rL42w_cSNV_post_lib_by_cadd_cor_s
BRCA1regionrL41rL42a_tSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41rL42w_tSNV_post_pre_by_cadd_cor_s
BRCA1regionrL41rL42a_tSNV_post_lib_by_cadd_cor_s
BRCA1regionrL41rL42a_tSNV_pre_lib_by_cadd_cor_s
```
Lollipop Plots
--------------
```{r}
#post_pre_mean by chr_pos with consequence, letter (genomic+), no legend.
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) + 
             as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = log2(post_pre_ratio_rL41rL42_mean_synnorm),
                   xend = as.numeric(as.character(pos)) +
                     as.numeric(as.character(pos_nudge)),
                   colour = conseq),
               yend = 0) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1.3) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "none") +
  ylab('Log-2 score') + 
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) + 
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```

```{r}
#tHDR_post_pre_mean by chr_pos with consequence, letter (genomic+), no legend.
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df, 
       aes(y = log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) +
             as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                   xend = as.numeric(as.character(pos)) +
                     as.numeric(as.character(pos_nudge)),
                   colour = conseq),
               yend = 0) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.3) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title=element_blank(),
        legend.key = element_blank(),
        legend.position = "none") + 
  ylab('Log-2 score') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5, 
            vjust = 0.5, 
            size = 1.4,
            colour='White')
```

```{r}
#instead post_lib_ratio:
#post_lib_mean by chr_pos with consequence, letter (genomic+), no legend.
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = log2(post_lib_ratio_rL41rL42_mean_synnorm),
                   xend = as.numeric(as.character(pos)) + 
                     as.numeric(as.character(pos_nudge)),
                   colour = conseq),
               yend = 0) + 
  geom_point(aes(colour = conseq), 
             alpha = 1, 
             size = 1.3) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "none") +
  ylab('Log-2 score') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```

```{r}
#tHDR_post_lib_mean by chr_pos with consequence, letter (genomic+), no legend.
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
                   xend = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)),
                   colour = conseq),
               yend = 0) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.3) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "none") +
  ylab('Log-2 score') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) + 
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```

```{r}
#CADD scores by position:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = as.numeric(as.character(CADD.phred)),
           x = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
                   xend = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)),
                   colour = conseq),
               yend = 0) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1.3) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "none") +
  ylab('Log-2 score') +
  xlab('Chr. 17 position') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```

```{r}
#Post_pre_ratio by position with clinvar line, conseq, and letter
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) +
             as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = post_pre_ratio_rL41rL42_mean_synnorm,
                   xend = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)),
                   colour = clinvar_simple),
               yend = 0) +
  geom_point(aes(colour = conseq),
             alpha = 1, 
             size = 1.3) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(), 
        legend.key = element_blank()) + 
  ylab('Log-2 score') +
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```

```{r}
#Clinvar pos without letter or legend:
ggplot(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
       aes(y = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           x = as.numeric(as.character(pos)) + as.numeric(as.character(pos_nudge)))) +
  geom_segment(aes(yend = post_pre_ratio_rL41rL42_mean_synnorm,
                   xend = as.numeric(as.character(pos)) +
                     as.numeric(as.character(pos_nudge)),
                   colour = clinvar_simple),
               yend = 0) +
  geom_point(aes(colour = conseq),
             alpha = 1, 
             size = 1.3) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'None') +
  ylab('Log-2 score') + 
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple) +
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')
```
Histograms of prog_map_cSNVs in lib, pre, post
----------------------------------------------
```{r}
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = lib_freq)) +
  geom_histogram(bins = 50,
                 fill = 'DarkGreen') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'none') +
  ylab('Single Nucleotide Variants') + 
  xlab('Reads') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank())
```

```{r}
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = pre_freq)) +
  geom_histogram(bins = 50,
                 fill = 'Green4') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'none') +
  ylab('Single Nucleotide Variants') + 
  xlab('Reads') + 
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank())
```

```{r}
ggplot(BRCA1regionrL41_prog_map_thresh5_df,
       aes(x = post_freq)) +
  geom_histogram(bins = 50,
                 fill = 'Green2') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'none') +
  ylab('Single Nucleotide Variants') + 
  xlab('Reads') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank())
```

```{r}
#scores correlate to CADD
ggplot(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df,
       aes(x = log2(post_pre_ratio_rL41rL42_mean_synnorm),
           y = as.numeric(as.character(CADD.phred)))) +
  geom_point(aes(colour = conseq),
             alpha = 1,
             size = 1) +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'none') +
  xlab('Enrichment (Log-2)') +
  ylab('CADD Score') +
  theme(panel.background = element_rect(fill = 'white',
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) +
  scale_fill_manual(values = named_conseq_colors_full_simple) +
  scale_color_manual(values = named_conseq_colors_full_simple)
```
Correlations for Heatmapping
------------
```{r}
#final correlations to save as variables for heatmapping:
BRCA1regionrL41_cSNV_pre_lib_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$lib_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_thresh5_df$pre_pseudo_freq),
                                   method = 'spearman')
BRCA1regionrL41_tSNV_pre_lib_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_lib_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_pre_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41_cSNV_post_lib_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$lib_pseudo_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df$post_pseudo_freq),
                                    method='spearman')
BRCA1regionrL41_tSNV_post_lib_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_lib_pseudo_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_post_pseudo_freq),
                                    method='spearman')
BRCA1regionrL41_cSNV_post_pre_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$post_pseudo_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df$pre_pseudo_freq),
                                    method='spearman')
BRCA1regionrL41_tSNV_post_pre_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_post_pseudo_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df$tHDR_pre_pseudo_freq),
                                    method='spearman')
BRCA1regionrL41_cSNV_rna_lib_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df$lib_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df$rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41_tSNV_rna_lib_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df$tHDR_lib_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df$tHDR_rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41_cSNV_rna_pre_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df$pre_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df$rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41_tSNV_rna_pre_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df$tHDR_rna_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df$tHDR_pre_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41rL42_cSNV_pre_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_freq),
                                log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_freq),
                                   method = 'spearman')
BRCA1regionrL41rL42_tSNV_pre_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_pre_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_pre_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41rL42_cSNV_pre_lib_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pre_lib_ratio),
                                             log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_pre_lib_ratio),
                                             method='spearman')
BRCA1regionrL41rL42_tSNV_pre_lib_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio),
                                             log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_pre_lib_ratio),
                                             method='spearman')
BRCA1regionrL41rL42_cSNV_post_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_freq),
                                    method='spearman')
BRCA1regionrL41rL42_tSNV_post_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pseudo_freq),
                                    log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pseudo_freq),
                                    method='spearman')
BRCA1regionrL41rL42_cSNV_post_lib_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_lib_ratio_synnorm),
                                              log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_lib_ratio_synnorm),
                                              method='spearman')
BRCA1regionrL41rL42_tSNV_post_lib_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_synnorm),
                                              log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_lib_ratio_synnorm),
                                              method='spearman')
BRCA1regionrL41rL42_cSNV_post_pre_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$post_pre_ratio_synnorm),
                                              log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_post_pre_ratio_synnorm),
                                              method='spearman')
BRCA1regionrL41rL42_tSNV_post_pre_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm),
                                              log2(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm),
                                              method='spearman')
BRCA1regionrL41rL42_cSNV_rna_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rna_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rL42_rna_freq),
                                   method='spearman')
BRCA1regionrL41rL42_tSNV_rna_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$tHDR_rna_pseudo_freq),
                                   log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rL42_tHDR_rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL41rL42_tSNV_rna_pre_ratio_cors = cor(log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$tHDR_rna_pre_ratio_synnorm),
                                             log2(BRCA1regionrL41_prog_map_exon_thresh5_df_rL42_int_ord_df$rL42_tHDR_rna_pre_ratio_synnorm),
                                             method='spearman')
BRCA1regionrL41_cSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$pre_lib_ratio),
                                              as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                              method='spearman')
BRCA1regionrL41_tSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_pre_lib_ratio),
                                              as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                              method='spearman')
BRCA1regionrL41_cSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$post_lib_ratio),
                                               as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL41_tSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_post_lib_ratio),
                                               as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL41_cSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$post_pre_ratio),
                                               as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL41_tSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_df$tHDR_post_pre_ratio),
                                               as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL41_cSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio),
                                                        as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                        method='spearman')
BRCA1regionrL41_tSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio),
                                                        as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                        method='spearman')
BRCA1regionrL41_cSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                            method='spearman')
BRCA1regionrL41_tSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                            method='spearman')
BRCA1regionrL41_cSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                            method='spearman')
BRCA1regionrL41_tSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                            method='spearman')
BRCA1regionrL41rL42a_cSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$pre_lib_ratio_rL41rL42_mean_synnorm),
                                                   as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                   method='spearman')
BRCA1regionrL41rL42a_tSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),
                                                   as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                   method='spearman')
BRCA1regionrL41rL42a_cSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_lib_ratio_rL41rL42_mean_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42a_tSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_pre_ratio_rL41rL42_mean_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio_rL41rL42_mean_synnorm),
                                                             as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                             method='spearman')
BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                                                             as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                             method='spearman')
BRCA1regionrL41rL42a_cSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio_rL41rL42_mean_synnorm),
                                                                 as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                                 method='spearman')
BRCA1regionrL41rL42a_tSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                                                                 as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                                 method='spearman')
BRCA1regionrL41rL42a_cSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$post_pre_ratio_rL41rL42_mean_synnorm),
                                                                 as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                                 method='spearman')
BRCA1regionrL41rL42a_tSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$tHDR_post_pre_ratio_rL41rL42_mean_synnorm),
                                                                 as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                                 method='spearman')
BRCA1regionrL41rL42w_cSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$pre_lib_ratio_rL41rL42_pre_weighted),
                                                   as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                   method='spearman')
BRCA1regionrL41rL42w_cSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_lib_ratio_rL41rL42_pre_weighted),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42w_cSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$post_pre_ratio_rL41rL42_pre_weighted_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')
BRCA1regionrL41rL42w_tSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_rL41rL42_pre_weighted_synnorm),
                                                    as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df$CADD.phred)),
                                                    method='spearman')

BRCA1regionrL41_cors <- c("BRCA1regionrL41",
                     BRCA1regionrL41_cSNV_pre_lib_cors,
                     BRCA1regionrL41_tSNV_pre_lib_cors,
                     BRCA1regionrL41_cSNV_post_lib_cors,
                     BRCA1regionrL41_tSNV_post_lib_cors,
                     BRCA1regionrL41_cSNV_post_pre_cors,
                     BRCA1regionrL41_tSNV_post_pre_cors,
                     BRCA1regionrL41_cSNV_rna_lib_cors,
                     BRCA1regionrL41_tSNV_rna_lib_cors,
                     BRCA1regionrL41_cSNV_rna_pre_cors,
                     BRCA1regionrL41_tSNV_rna_pre_cors,
                     BRCA1regionrL41rL42_cSNV_pre_cors,
                     BRCA1regionrL41rL42_tSNV_pre_cors,
                     BRCA1regionrL41rL42_cSNV_pre_lib_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_pre_lib_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_post_cors,
                     BRCA1regionrL41rL42_tSNV_post_cors,
                     BRCA1regionrL41rL42_cSNV_post_lib_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_post_lib_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_post_pre_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_post_pre_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_rna_cors,
                     BRCA1regionrL41rL42_tSNV_rna_cors,
                     BRCA1regionrL41rL42_tSNV_rna_pre_ratio_cors,
                     BRCA1regionrL41_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41_tSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41_tSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41_tSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41_cSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41_tSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41_cSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41_tSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41_cSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41_tSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42w_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_tSNV_post_pre_ratio_cadd_cors)
```
Replicate 2
```{r}
#replicate 2 assignments for rL42-specific scores...
BRCA1regionrL42_cSNV_pre_lib_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$lib_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_thresh5_df$pre_pseudo_freq),
                                   method='spearman')
BRCA1regionrL42_tSNV_pre_lib_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_lib_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_pre_pseudo_freq),
                                   method='spearman')
BRCA1regionrL42_cSNV_post_lib_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$lib_pseudo_freq),
                                    log2(BRCA1regionrL42_prog_map_thresh5_df$post_pseudo_freq),
                                    method='spearman')
BRCA1regionrL42_tSNV_post_lib_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_lib_pseudo_freq),
                                    log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_post_pseudo_freq),
                                    method='spearman')
BRCA1regionrL42_cSNV_post_pre_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$post_pseudo_freq),
                                    log2(BRCA1regionrL42_prog_map_thresh5_df$pre_pseudo_freq),
                                    method='spearman')
BRCA1regionrL42_tSNV_post_pre_cors = cor(log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_post_pseudo_freq),
                                    log2(BRCA1regionrL42_prog_map_thresh5_df$tHDR_pre_pseudo_freq),
                                    method='spearman')
BRCA1regionrL42_cSNV_rna_lib_cors = cor(log2(BRCA1regionrL42_prog_map_exon_thresh5_df$lib_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_exon_thresh5_df$rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL42_tSNV_rna_lib_cors = cor(log2(BRCA1regionrL42_prog_map_exon_thresh5_df$tHDR_lib_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_exon_thresh5_df$tHDR_rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL42_cSNV_rna_pre_cors = cor(log2(BRCA1regionrL42_prog_map_exon_thresh5_df$pre_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_exon_thresh5_df$rna_pseudo_freq),
                                   method='spearman')
BRCA1regionrL42_tSNV_rna_pre_cors = cor(log2(BRCA1regionrL42_prog_map_exon_thresh5_df$tHDR_rna_pseudo_freq),
                                   log2(BRCA1regionrL42_prog_map_exon_thresh5_df$tHDR_pre_pseudo_freq),
                                   method='spearman')
# rL42 switching to the df with CADD scores to make correlations with CADD
BRCA1regionrL42_cSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$pre_lib_ratio),
                                              as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                              method='spearman')
BRCA1regionrL42_tSNV_pre_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_pre_lib_ratio),
                                              as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                              method='spearman')
BRCA1regionrL42_cSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$post_lib_ratio),
                                               as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL42_tSNV_post_lib_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_post_lib_ratio),
                                               as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL42_cSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$post_pre_ratio),
                                               as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL42_tSNV_post_pre_ratio_cadd_cors = cor(log2(BRCA1regionrL42_prog_map_CADD_df$tHDR_post_pre_ratio),
                                               as.numeric(as.character(BRCA1regionrL42_prog_map_CADD_df$CADD.phred)),
                                               method='spearman')
BRCA1regionrL42_cSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_post_pre_ratio),
                                                        as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                        method='spearman')
BRCA1regionrL42_tSNV_post_pre_ratio_cadd_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_tHDR_post_pre_ratio),
                                                        as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$CADD.phred)),
                                                        method='spearman')
BRCA1regionrL42_cSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                            method='spearman')
BRCA1regionrL42_tSNV_post_pre_ratio_polyphen_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_tHDR_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$polyphen.val)),
                                                            method='spearman')
BRCA1regionrL42_cSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                            method='spearman')
BRCA1regionrL42_tSNV_post_pre_ratio_grantham_missense_cors = cor(log2(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$rL42_tHDR_post_pre_ratio),
                                                            as.numeric(as.character(BRCA1regionrL41_prog_map_CADD_thresh5_df_rL42_int_ord_df_missense$Grantham)),
                                                            method='spearman')

BRCA1regionrL42_cors <- c("BRCA1regionrL42",
                     BRCA1regionrL42_cSNV_pre_lib_cors,
                     BRCA1regionrL42_tSNV_pre_lib_cors,
                     BRCA1regionrL42_cSNV_post_lib_cors,
                     BRCA1regionrL42_tSNV_post_lib_cors,
                     BRCA1regionrL42_cSNV_post_pre_cors,
                     BRCA1regionrL42_tSNV_post_pre_cors,
                     BRCA1regionrL42_cSNV_rna_lib_cors,
                     BRCA1regionrL42_tSNV_rna_lib_cors,
                     BRCA1regionrL42_cSNV_rna_pre_cors,
                     BRCA1regionrL42_tSNV_rna_pre_cors,
                     BRCA1regionrL41rL42_cSNV_pre_cors,
                     BRCA1regionrL41rL42_tSNV_pre_cors,
                     BRCA1regionrL41rL42_cSNV_pre_lib_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_pre_lib_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_post_cors,
                     BRCA1regionrL41rL42_tSNV_post_cors,
                     BRCA1regionrL41rL42_cSNV_post_lib_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_post_lib_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_post_pre_ratio_cors,
                     BRCA1regionrL41rL42_tSNV_post_pre_ratio_cors,
                     BRCA1regionrL41rL42_cSNV_rna_cors,
                     BRCA1regionrL41rL42_tSNV_rna_cors,
                     BRCA1regionrL41rL42_tSNV_rna_pre_ratio_cors,
                     BRCA1regionrL42_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL42_tSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL42_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL42_tSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL42_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL42_tSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL42_cSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL42_tSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL42_cSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL42_tSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL42_cSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL42_tSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_cadd_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_polyphen_missense_cors,
                     BRCA1regionrL41rL42a_cSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42a_tSNV_post_pre_ratio_grantham_missense_cors,
                     BRCA1regionrL41rL42w_cSNV_pre_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_cSNV_post_lib_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_cSNV_post_pre_ratio_cadd_cors,
                     BRCA1regionrL41rL42w_tSNV_post_pre_ratio_cadd_cors)
```

```{r}
#clear the large data frames from memory
rm(BRCA1regionrL41)
rm(BRCA1regionrL42)
gc()
```
Positional modeling included here.
Requires exon snv dataframes to be present.
Note that this averages replicates 1,2 after fitting a model for each independently.
For fitting the model, more lax criteria for inclusion were used due to fewer LOF variants needing to be excluded. (0.3 pre-lib and 0.7 post-pre).
```{r}
#a function to extend the lowess fit to points missing on either end of the sequence (must be used on an ordered sequence vector, as supplied here. Will only fill in missing points on the ends)
library(gridExtra)
extend_fit <- function(fit_vector) {
  if (length(which(is.na(fit_vector))) == 0) {
    out_vector <- fit_vector}
  else {
    min_na <- min(which(is.na(fit_vector) == TRUE))
    min_point <- min(which(is.na(fit_vector) == FALSE))
    min_point_val <- fit_vector[min_point]
    max_point <- max(which(is.na(fit_vector) == FALSE))
    max_point_val <- fit_vector[max_point]
    max_na <- max(which(is.na(fit_vector) == TRUE))
    out_vector <- fit_vector
    if (min_na == 1) {
      out_vector[min_na:min_point] <- min_point_val}
    if (max_na == length(fit_vector)) {
      out_vector[(max_point + 1):length(fit_vector)] <- max_point_val}}
  return(out_vector)}
```
BRCA1regionrL41
```{r}
BRCA1region_posrange <- range(as.numeric(as.character(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$pos)))

BRCA1region_posseq <- seq(from = BRCA1region_posrange[1], to = BRCA1region_posrange[2])

BRCA1regionrL41_tHDR_pre_lib_ratio.lo <- loess(log2(tHDR_pre_lib_ratio_synnorm) ~ as.numeric(as.character(pos)),
                                          BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
                                          subset = which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_synnorm > 0.7 &
        BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio_synnorm > 0.3),
        model = TRUE,
        span = 0.15)

BRCA1regionrL41_tHDR_pre_lib_pred <- predict(BRCA1regionrL41_tHDR_pre_lib_ratio.lo,
                                        newdata = BRCA1region_posseq,
                                        se = TRUE)
BRCA1regionrL41_tHDR_pre_lib_pos_effects <- BRCA1regionrL41_tHDR_pre_lib_pred$fit

#.x represents the extended pos_effects.
BRCA1regionrL41_tHDR_pre_lib_pos_effects.x <- extend_fit(BRCA1regionrL41_tHDR_pre_lib_pos_effects)

BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci <- BRCA1regionrL41_tHDR_pre_lib_pred$se.fit * qt(0.95 / 2 + .5, BRCA1regionrL41_tHDR_pre_lib_pred$df)

BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci_ymin = BRCA1regionrL41_tHDR_pre_lib_pos_effects - BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci_ymax = BRCA1regionrL41_tHDR_pre_lib_pos_effects + BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL41_tHDR_pre_lib_pos_effects.df <- data.frame(pos = BRCA1region_posseq,
                                                     BRCA1regionrL41_tHDR_pre_lib_pos_effects,
                                                     BRCA1regionrL41_tHDR_pre_lib_pos_effects.x,
                                                     BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci_ymin,
                                                     BRCA1regionrL41_tHDR_pre_lib_pos_effects_ci_ymax,
                                                     BRCA1regionrL41_tHDR_pre_lib_pred_se = BRCA1regionrL41_tHDR_pre_lib_pred$se.fit)
```
BRCA1regionrL42
```{r}
BRCA1regionrL42_tHDR_pre_lib_ratio.lo <- loess(log2(rL42_tHDR_pre_lib_ratio_synnorm) ~ as.numeric(as.character(pos)),
                                          BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
                                          subset = which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_post_pre_ratio_synnorm > 0.7 &
        BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$rL42_tHDR_pre_lib_ratio_synnorm > 0.3),
        model = TRUE,
        span = 0.15)

BRCA1regionrL42_tHDR_pre_lib_pred <- predict(BRCA1regionrL42_tHDR_pre_lib_ratio.lo,
                                        newdata = BRCA1region_posseq,
                                        se = TRUE)

BRCA1regionrL42_tHDR_pre_lib_pos_effects <- BRCA1regionrL42_tHDR_pre_lib_pred$fit

#.x represents the extended pos_effects.
BRCA1regionrL42_tHDR_pre_lib_pos_effects.x <- extend_fit(BRCA1regionrL42_tHDR_pre_lib_pos_effects)

BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci <- BRCA1regionrL42_tHDR_pre_lib_pred$se.fit * qt(0.95 / 2 + .5, BRCA1regionrL42_tHDR_pre_lib_pred$df)

BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci_ymin = BRCA1regionrL42_tHDR_pre_lib_pos_effects - BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci_ymax = BRCA1regionrL42_tHDR_pre_lib_pos_effects + BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL42_tHDR_pre_lib_pos_effects.df <- data.frame(pos = BRCA1region_posseq,
                                                     BRCA1regionrL42_tHDR_pre_lib_pos_effects,
                                                     BRCA1regionrL42_tHDR_pre_lib_pos_effects.x,
                                                     BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci_ymin,
                                                     BRCA1regionrL42_tHDR_pre_lib_pos_effects_ci_ymax,
                                                     BRCA1regionrL42_tHDR_pre_lib_pred_se = BRCA1regionrL42_tHDR_pre_lib_pred$se.fit)
```
BRCA1regionrL41rL42
```{r}
BRCA1regionrL41rL42_tHDR_pre_lib_ratio.lo <- loess(log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm) ~ as.numeric(as.character(pos)),
                                              BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df,
                                              subset = which(BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_post_pre_ratio_rL41rL42_mean_synnorm > 0.7 &
        BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df$tHDR_pre_lib_ratio_rL41rL42_mean_synnorm > 0.3),
        model = TRUE,
        span = 0.15)

BRCA1regionrL41rL42_tHDR_pre_lib_pred <- predict(BRCA1regionrL41rL42_tHDR_pre_lib_ratio.lo,
                                            newdata = BRCA1region_posseq,
                                            se = TRUE)

BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects <- BRCA1regionrL41rL42_tHDR_pre_lib_pred$fit

#.x represents the extended pos_effects.
BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x <- extend_fit(BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects)

BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci <- BRCA1regionrL41rL42_tHDR_pre_lib_pred$se.fit * qt(0.95 / 2 + .5, BRCA1regionrL41rL42_tHDR_pre_lib_pred$df)

BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci_ymin = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects - BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci_ymax = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects + BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci

BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df <- data.frame(pos = BRCA1region_posseq,
                                                         BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects,
                                                         BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x,
                                                         BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci_ymin,
                                                         BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects_ci_ymax,
                                                         BRCA1regionrL41rL42_tHDR_pre_lib_pred_se = BRCA1regionrL41rL42_tHDR_pre_lib_pred$se.fit)
```
Merge Data Frames
----------------
```{r}
#merge into a single df (merge original df, and pos effects (extended) df's from all 3 fits -- rL41, rL42, rL41rL42)
BRCA1regionrL4_tHDR_pos_merge_df <- BRCA1regionrL41_prog_map_thresh5_df_rL42_int_ord_df 

BRCA1regionrL4_tHDR_pos_merge_df <- merge(BRCA1regionrL4_tHDR_pos_merge_df,
                                     BRCA1regionrL41_tHDR_pre_lib_pos_effects.df[, c("pos", "BRCA1regionrL41_tHDR_pre_lib_pos_effects.x")],
                                     by="pos",
                                     all.x=TRUE)

BRCA1regionrL4_tHDR_pos_merge_df <- merge(BRCA1regionrL4_tHDR_pos_merge_df,
                                     BRCA1regionrL42_tHDR_pre_lib_pos_effects.df[, c("pos","BRCA1regionrL42_tHDR_pre_lib_pos_effects.x")],
                                     by="pos",
                                     all.x=TRUE)

#for rL42, need post-lib and pre-lib values first
BRCA1regionrL4_tHDR_pos_merge_df <- merge(BRCA1regionrL4_tHDR_pos_merge_df,
                                     BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df[, c("pos","BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x")],
                                     by="pos",
                                     all.x=TRUE)
```

```{r}
#Calculate adjusted ratios and put into df (along with affect sizes)
#5 adjustments for each post-lib and pre-lib measurement (e.g. rL41 vs. rL41_fit and vs. mean_fit)
BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_post_lib_loess_rL41 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_post_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_post_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_post_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_post_lib_loess_rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$rL42_tHDR_post_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_post_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$rL42_tHDR_post_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_post_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_post_lib_ratio_rL41rL42_mean_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x

###same for pre-lib
BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_pre_lib_loess_rL41 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_pre_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_pre_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_pre_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_pre_lib_loess_rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$rL42_tHDR_pre_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_pre_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$rL42_tHDR_pre_lib_ratio_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x

BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_loess_rL41rL42 <- log2(BRCA1regionrL4_tHDR_pos_merge_df$tHDR_pre_lib_ratio_rL41rL42_mean_synnorm) - BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x
```
Position Correction Plots
----------------
```{r}
#these plots have set cartesians for comparing on same axes -- plot to illustrate how pos-correction works.
#plotting rL41:  original data, fit 1, and fit 2, for tHDR_pre_lib
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = log2(tHDR_pre_lib_ratio_synnorm),
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq,
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
    coord_cartesian(ylim = c(-3, 3)) +
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") +
      ylab('pre / lib (log-2)') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) +
      scale_color_manual(values = named_conseq_colors) +
      geom_vline(xintercept = BRCA1region_cut_pos,
                 lty = 2) +
      geom_hline(yintercept = 0,
                 lty = 3) +
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White') +
      geom_line(aes(y = BRCA1regionrL41_tHDR_pre_lib_pos_effects.x,
                    colour = 'REF',
                    size = 0.5,
                    alpha = 0.3),
                data = BRCA1regionrL41_tHDR_pre_lib_pos_effects.df,
                stat = "identity") +
      geom_line(aes(y = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x,
                    colour = 'SYNONYMOUS',
                    size = 0.5,
                    alpha = 0.3),
      data = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df,
      stat = "identity"), 
    ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
           aes(y = BRCA1regionrL41_tHDR_pre_lib_loess_rL41,
               x = as.numeric(as.character(pos)))) +
    geom_point(aes(colour = conseq,
                   size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
      coord_cartesian(ylim = c(-3, 3)) +
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") +
      ylab('pre / lib cor.rL41 ') +
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) +
      scale_color_manual(values = named_conseq_colors) +
      geom_vline(xintercept = BRCA1region_cut_pos,
                 lty = 2) +
      geom_hline(yintercept = 0,
                 lty = 3) +
      geom_text( aes(label = alt),
                 hjust = 0.5,
                 vjust = 0.5,
                 size = 1.4,
                 colour = 'White'), 
    ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
           aes(y = BRCA1regionrL41_tHDR_pre_lib_loess_rL41rL42,
               x = as.numeric(as.character(pos)))) +
    geom_point(aes(colour = conseq, 
                   size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
      coord_cartesian(ylim = c(-3, 3)) +
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") +
      ylab('pre / lib cor.mean') +
      xlab('Chr. 17 position') +
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
      theme(text = element_text(size = 12),
            axis.text.y = element_text(size = 12),
            axis.text.x = element_text(size = 12),
            strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) +
      scale_color_manual(values = named_conseq_colors) +
      geom_vline(xintercept = BRCA1region_cut_pos,
                 lty = 2) +
      geom_hline(yintercept = 0, 
                 lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White'),
    nrow = 3)
```

```{r}
#redo same plot but with rL42:
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = log2(rL42_tHDR_pre_lib_ratio_synnorm),
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq,
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
               coord_cartesian(ylim = c(-3,3)) +
               theme(panel.background = element_rect(fill = 'white',
                                                     colour = 'white')) +
               theme(legend.title=element_blank(),
                     legend.key = element_blank(),
                     legend.position = "None") +
               ylab('pre / lib (log-2)') +
               xlab('Chr. 17 position') +
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) +
               scale_fill_manual(values = named_conseq_colors) +
               scale_color_manual(values = named_conseq_colors) +
               geom_vline(xintercept = BRCA1region_cut_pos,
                          lty = 2) +
               geom_hline(yintercept = 0,
                          lty = 3) +
               geom_text(aes(label = alt),
                         hjust = 0.5, 
                         vjust = 0.5, 
                         size = 1.4,
                         colour ='White') +
               geom_line(aes(y = BRCA1regionrL42_tHDR_pre_lib_pos_effects.x,
                             colour = 'REF',
                             size = 0.5,
                             alpha = 0.3), 
                         data = BRCA1regionrL42_tHDR_pre_lib_pos_effects.df, 
                         stat = "identity") +
               geom_line(aes(y = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x,
                             colour = 'SYNONYMOUS',
                             size = 0.5,
                             alpha = 0.3), 
                         data = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df, 
                         stat = "identity"), 
             ggplot(BRCA1regionrL4_tHDR_pos_merge_df, 
                    aes(y = BRCA1regionrL42_tHDR_pre_lib_loess_rL42,
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq,
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
               coord_cartesian(ylim = c(-3,3)) +
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),
                                                                              legend.position = "None") +
               ylab('pre / lib cor.rL42 ') +
               xlab('Chr. 17 position') +
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) + 
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) +
               scale_fill_manual(values = named_conseq_colors) +
               scale_color_manual(values = named_conseq_colors) +
               geom_vline(xintercept = BRCA1region_cut_pos,
                          lty = 2) +
               geom_hline(yintercept = 0,
                          lty = 3) +
               geom_text(aes(label = alt),
                         hjust = 0.5, 
                         vjust = 0.5, 
                         size = 1.4,
                         colour = 'White'), 
             ggplot(BRCA1regionrL4_tHDR_pos_merge_df, 
                    aes(y = BRCA1regionrL42_tHDR_pre_lib_loess_rL41rL42,
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq,
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) + 
               coord_cartesian(ylim = c(-3, 3)) +
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(legend.title = element_blank(),
                     legend.key = element_blank(),
                     legend.position = "None") + 
               ylab('pre / lib cor.mean') +
               xlab('Chr. 17 position') + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) + 
               scale_fill_manual(values = named_conseq_colors) + 
               scale_color_manual(values = named_conseq_colors) + 
               geom_vline(xintercept = BRCA1region_cut_pos, 
                          lty = 2) +
               geom_hline(yintercept = 0, 
                          lty = 3) + 
               geom_text(aes(label = alt),
                         hjust = 0.5,
                         vjust = 0.5,
                         size = 1.4,
                         colour = 'White'),
             nrow = 3)
```

```{r}
#plot of the averaged data with all 3 fit lines (SYN is average, two indiv. reps in gray), and then corrected according to averaged fit (two plots)
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df, 
                    aes(y = log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq, 
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
               coord_cartesian(ylim = c(-3, 3)) +
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(legend.title = element_blank(),
                     legend.key = element_blank(),
                     legend.position = "None") + 
               ylab('pre / lib (log-2)') + 
               xlab('Chr. 17 position') + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) + 
               scale_fill_manual(values = named_conseq_colors) + 
               scale_color_manual(values = named_conseq_colors) +
               geom_vline(xintercept = BRCA1region_cut_pos, 
                          lty = 2) +
               geom_hline(yintercept = 0, 
                          lty = 3) + 
               geom_text(aes(label = alt),
                         hjust = 0.5,
                         vjust = 0.5,
                         size = 1.4,
                         colour = 'White') + 
               geom_line(aes(y = BRCA1regionrL41_tHDR_pre_lib_pos_effects.x,
                             colour = 'REF',
                             size = 0.5,
                             alpha = 0.3),
                         data = BRCA1regionrL41_tHDR_pre_lib_pos_effects.df,
                         stat = "identity") + 
               geom_line(aes(y = BRCA1regionrL42_tHDR_pre_lib_pos_effects.x,
                             colour = 'REF',
                             size = 0.5,
                             alpha = 0.3),
                         data = BRCA1regionrL42_tHDR_pre_lib_pos_effects.df,
                         stat = "identity") +
               geom_line(aes(y = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x,
                             colour = 'SYNONYMOUS',
                             size = 0.5,
                             alpha = 0.3),
                         data = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df,
                         stat = "identity"), 
             ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = BRCA1regionrL41rL42_tHDR_pre_lib_loess_rL41rL42,
                        x = as.numeric(as.character(pos)))) + 
               geom_point(aes(colour = conseq, 
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
               coord_cartesian(ylim = c(-3, 3)) + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(legend.title = element_blank(),
                     legend.key = element_blank(),
                     legend.position = "None") +
               ylab('pre / lib cor.rL41rL42') +
               xlab('Chr. 17 position') +
               theme(panel.background = element_rect(fill = 'white',
                                                     colour = 'white')) +
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) + 
                 scale_fill_manual(values = named_conseq_colors) + 
                 scale_color_manual(values = named_conseq_colors) + 
                 geom_vline(xintercept = BRCA1region_cut_pos, 
                            lty = 2) +
                 geom_hline(yintercept = 0, 
                            lty = 3) +
                 geom_text(aes(label = alt),
                           hjust = 0.5,
                           vjust = 0.5,
                           size = 1.4,
                           colour = 'White',
                           nrow = 2))
```

```{r}
#plotting original post-lib data (mean), w/ all 3 fits, and subtracted post-lib data using the averages fit.
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm),
                        x = as.numeric(as.character(pos)))) +
               geom_point(aes(colour = conseq, 
                              size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
               coord_cartesian(ylim = c(-6, 3)) +
               theme(panel.background = element_rect(fill = 'white',
                                                     colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") +
      ylab('post / lib (log-2)') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) +
      scale_color_manual(values = named_conseq_colors) +
      geom_vline(xintercept = BRCA1region_cut_pos,
                 lty = 2) +
      geom_hline(yintercept = 0, 
                 lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White') +
      geom_line(
        aes(y = BRCA1regionrL41_tHDR_pre_lib_pos_effects.x,
            colour = 'REF',
            size = 0.5,
            alpha = 0.3),
        data = BRCA1regionrL41_tHDR_pre_lib_pos_effects.df,
        stat = "identity") +
      geom_line(aes(y = BRCA1regionrL42_tHDR_pre_lib_pos_effects.x,
                    colour = 'REF',
                    size = 0.5,
                    alpha = 0.3), 
                data = BRCA1regionrL42_tHDR_pre_lib_pos_effects.df,
                stat = "identity") + 
      geom_line(aes(y = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.x,
                    colour = 'SYNONYMOUS',
                    size = 0.5,
                    alpha = 0.3), 
                data = BRCA1regionrL41rL42_tHDR_pre_lib_pos_effects.df,
                stat = "identity"), 
    ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
           aes(y = BRCA1regionrL41rL42_tHDR_post_lib_loess_rL41rL42,
               x = as.numeric(as.character(pos)))) +
      geom_point(aes(colour = conseq, 
                     size = tHDR_post_pre_ratio_rL41rL42_mean_synnorm)) +
      coord_cartesian(ylim = c(-6, 3)) +
      theme(panel.background = element_rect(fill = 'white',
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") +
      ylab('post / lib cor.rL41rL42 ') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) +
      scale_color_manual(values = named_conseq_colors) + 
      geom_vline(xintercept = BRCA1region_cut_pos, 
                 lty = 2) +
      geom_hline(yintercept = 0, 
                 lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White'),
    nrow = 2)
```

```{r}
#plot the 2 independently processed replicates against each other:
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = BRCA1regionrL41_tHDR_post_lib_loess_rL41, 
                        x = as.numeric(as.character(pos)))) + 
               geom_point(aes(colour = conseq, 
                              size = tHDR_post_pre_ratio_synnorm)) +
               coord_cartesian(ylim = c(-6, 3)) + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") + 
      ylab('rL41 post / lib cor.') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) + 
      scale_fill_manual(values = named_conseq_colors) + 
      scale_color_manual(values = named_conseq_colors) + 
      geom_vline(xintercept = BRCA1region_cut_pos, 
                 lty = 2) +
      geom_hline(yintercept = 0, 
                 lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White'), 
  ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
         aes(y = BRCA1regionrL42_tHDR_post_lib_loess_rL42, 
             x = as.numeric(as.character(pos)))) + 
    geom_point(aes(colour = conseq, 
                   size = tHDR_post_pre_ratio_synnorm)) +
    coord_cartesian(ylim = c(-6, 3)) + 
    theme(panel.background = element_rect(fill = 'white', 
                                          colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") + 
    ylab('rL42 post / lib cor.') + 
    xlab('Chr. 17 position') + 
    theme(panel.background = element_rect(fill = 'white', 
                                          colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) + 
    scale_fill_manual(values = named_conseq_colors) + 
    scale_color_manual(values = named_conseq_colors) +
    geom_vline(xintercept = BRCA1region_cut_pos, 
               lty = 2) +
    geom_hline(yintercept = 0, 
               lty = 3) + 
    geom_text(aes(label = alt),
              hjust = 0.5,
              vjust = 0.5,
              size = 1.4,
              colour = 'White'),
  nrow = 2)
```

```{r}
#test bimodality with a histogram (using each rep's post-lib w. each rep's fit):
BRCA1regionrL41_tHDR_post_lib_lo_histo <- ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                                            aes(x = BRCA1regionrL41_tHDR_post_lib_loess_rL41, 
                                                fill = conseq)) + 
  geom_histogram(alpha = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'None') + 
  ylab('SNVs') + 
  xlab('post / lib') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  coord_cartesian(xlim = c(-6, 2))

BRCA1regionrL42_tHDR_post_lib_lo_histo <- ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                                            aes(x = BRCA1regionrL42_tHDR_post_lib_loess_rL42, 
                                                fill = conseq)) + 
  geom_histogram(alpha = 1) + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = 'None') +
  ylab('SNVs') + 
  xlab('post / lib') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  coord_cartesian(xlim = c(-6, 2))

BRCA1regionrL41rL42_tHDR_post_lib_lo_histo = grid.arrange(BRCA1regionrL41_tHDR_post_lib_lo_histo,BRCA1regionrL42_tHDR_post_lib_lo_histo,nrow = 2)

BRCA1regionrL41rL42_tHDR_post_lib_lo_histo
```

```{r}
#Superceding prior normalization with simple normalization to mean for regions with few variants of high effect....
#include a within-replicate sns normalization (e.sns) next
#general formula:  b/(a-c)*(A-C) where b is snv, a is exon med syn, c is exon med nonsense, A is global med syn, C is global med nonsense.
#defines a for rL41
BRCA1regionrL4_tHDR_post_lib_loess_median_rL41 <- median(BRCA1regionrL4_tHDR_pos_merge_df[which(BRCA1regionrL4_tHDR_pos_merge_df$conseq != 'not_a_conseq'),c('BRCA1regionrL41_tHDR_post_lib_loess_rL41')])
#defines a for rL42
BRCA1regionrL4_tHDR_post_lib_loess_median_rL42 <- median(BRCA1regionrL4_tHDR_pos_merge_df[which(BRCA1regionrL4_tHDR_pos_merge_df$conseq != 'not_a_conseq'),c('BRCA1regionrL42_tHDR_post_lib_loess_rL42')])

#simply subtracting mean post-lib score for each
#calculate new b values for rL41 'e.sns' = exon syn/ns norm.
BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_post_lib_loess_rL41_e.sns <- (BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_post_lib_loess_rL41-BRCA1regionrL4_tHDR_post_lib_loess_median_rL41)
#calculate new b values for rL42 'e.sns' = exon syn/ns norm.
BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_post_lib_loess_rL42_e.sns <- (BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_post_lib_loess_rL42-BRCA1regionrL4_tHDR_post_lib_loess_median_rL42)

#calculate a new post_lib mean based on individually fit, exon-normalized data
BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean <- (BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL41_tHDR_post_lib_loess_rL41_e.sns+BRCA1regionrL4_tHDR_pos_merge_df$BRCA1regionrL42_tHDR_post_lib_loess_rL42_e.sns)/2

BRCA1regionrL4_tHDR_post_lib_loess_e.sns_rL41rL42_median <- median(BRCA1regionrL4_tHDR_pos_merge_df[which(BRCA1regionrL4_tHDR_pos_merge_df$conseq != 'not_a_conseq'),c('BRCA1regionrL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])

#plot the mean score from 2 replicates below each replicate:
grid.arrange(ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = BRCA1regionrL41_tHDR_post_lib_loess_rL41, 
                        x = as.numeric(as.character(pos)))) + 
               geom_point(aes(colour = conseq, 
                              size = tHDR_post_pre_ratio_synnorm)) +
               coord_cartesian(ylim = c(-6, 3)) + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(legend.title = element_blank(),
                     legend.key = element_blank(),
                     legend.position = "None") + 
               ylab('rL41 post / lib cor.') + 
               xlab('Chr. 17 position') + 
               theme(panel.background = element_rect(fill = 'white', 
                                                     colour = 'white')) +
               theme(text = element_text(size = 12),
                     axis.text.y = element_text(size = 12),
                     axis.text.x = element_text(size = 12),
                     strip.background = element_blank()) + 
               scale_fill_manual(values = named_conseq_colors) + 
               scale_color_manual(values = named_conseq_colors) + 
               geom_vline(xintercept = BRCA1region_cut_pos, 
                          lty = 2) +
               geom_hline(yintercept = 0, 
                          lty = 3) + 
               geom_text(aes(label = alt),
                         hjust = 0.5,
                         vjust = 0.5,
                         size = 1.4,
                         colour = 'White'), 
             ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
                    aes(y = BRCA1regionrL42_tHDR_post_lib_loess_rL42, 
                        x = as.numeric(as.character(pos)))) + 
    geom_point(aes(colour = conseq, 
                   size = tHDR_post_pre_ratio_synnorm)) +
    coord_cartesian(ylim = c(-6, 3)) + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") + 
      ylab('rL42 post / lib cor.') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) + 
      scale_fill_manual(values = named_conseq_colors) + 
      scale_color_manual(values = named_conseq_colors) + 
      geom_vline(xintercept = BRCA1region_cut_pos, 
                 lty = 2) +
    geom_hline(yintercept = 0, 
               lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White'), 
    ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
           aes(y = BRCA1regionrL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,
               x = as.numeric(as.character(pos)))) +
      geom_point(aes(colour = conseq, 
                     size = tHDR_post_pre_ratio_synnorm)) +
      coord_cartesian(ylim = c(-6, 3)) + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(legend.title = element_blank(),
          legend.key = element_blank(),
          legend.position = "None") + 
      ylab('rL42 post / lib cor.') + 
      xlab('Chr. 17 position') + 
      theme(panel.background = element_rect(fill = 'white', 
                                            colour = 'white')) +
    theme(text = element_text(size = 12),
          axis.text.y = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          strip.background = element_blank()) +
      scale_fill_manual(values = named_conseq_colors) + 
      scale_color_manual(values = named_conseq_colors) + 
      geom_vline(xintercept = BRCA1region_cut_pos, 
                 lty = 2) +
      geom_hline(yintercept = 0, 
                 lty = 3) + 
      geom_text(aes(label = alt),
                hjust = 0.5,
                vjust = 0.5,
                size = 1.4,
                colour = 'White'),
    nrow = 3)
```

```{r}
#plot the mean score of 2 replicates on it's own, with LOF score cut-off from exons
BRCA1regionrL4_mean_post_lib_cor_vs_pos = ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
       aes(y = BRCA1regionrL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, 
           x = as.numeric(as.character(pos)))) + 
  geom_point(aes(colour = conseq)) + 
  coord_cartesian(ylim = c(-6, 2)) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "None") + 
  ylab('mean post / lib cor.') + 
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  geom_vline(xintercept = BRCA1region_cut_pos, lty = 2) +
  #geom_hline(yintercept = 0, lty = 3) + geom_hline(yintercept = 0, lty = 3) +
  #geom_hline(yintercept = -1.328, lty = 2) + geom_hline(yintercept = -0.748, lty = 2) + 
  geom_text(aes(label = alt),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')

BRCA1regionrL4_mean_post_lib_cor_vs_pos
#ggsave("_pos.png", BRCA1regionrL4_mean_post_lib_cor_vs_pos, height = 4, width = 8)
```

```{r}
#mean function score (post/pre)
BRCA1regionrL4_mean_post_pre_vs_pos = ggplot(BRCA1regionrL4_tHDR_pos_merge_df,
       aes(y = log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm), 
           x = as.numeric(as.character(pos)))) + 
  geom_point(aes(colour = conseq)) + 
  coord_cartesian(ylim = c(-5, 1)) +
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "None") + 
  ylab(paste0(SGE_region, ' mean function score (post/pre)')) + 
  xlab('Chr. 17 position') + 
  theme(panel.background = element_rect(fill = 'white', 
                                        colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()) + 
  scale_fill_manual(values = named_conseq_colors) + 
  scale_color_manual(values = named_conseq_colors) + 
  scale_x_reverse() +
 # geom_vline(xintercept = BRCA1region_cut_pos, lty = 2) +
  #geom_hline(yintercept = 0, lty = 3) + geom_hline(yintercept = 0, lty = 3) +
  #geom_hline(yintercept = -1.328, lty = 2) + geom_hline(yintercept = -0.748, lty = 2) + 
  geom_text(aes(label = rev_comp),
            hjust = 0.5,
            vjust = 0.5,
            size = 1.4,
            colour = 'White')

BRCA1regionrL4_mean_post_pre_vs_pos


```


```{r}
#### saving ####
#
ggsave(paste0("plots/", date, "_", SGE_region, "_post_pre_by_pos_reversed.png"), BRCA1regionrL4_mean_post_pre_vs_pos, height = 4, width = 8)

#
write.csv(x = BRCA1regionrL4_tHDR_pos_merge_df, file = paste0("dfs/", date, "_BRCA1", SGE_region, "_tHDR_pos_merge_df.csv"), row.names = FALSE)


```


